{
 "cells": [
  {
   "metadata": {},
   "cell_type": "code",
   "source": [
    "import pandas as pd\n",
    "from pathlib import Path\n",
    "from drevalpy.datasets.curvecurator import preprocess, _exec_curvecurator, postprocess\n",
    "import drevalpy\n",
    "drevalpy.__version__"
   ],
   "id": "c1b17b1316171b42",
   "outputs": [],
   "execution_count": null
  },
  {
   "metadata": {},
   "cell_type": "markdown",
   "source": "## Load Data",
   "id": "f20b86ec5c732c42"
  },
  {
   "metadata": {},
   "cell_type": "code",
   "source": [
    "RAW_FILE_PATH = Path(\"beataml_wv1to4_raw_inhibitor_v4_dbgap.txt\")\n",
    "raw_df = pd.read_csv(RAW_FILE_PATH, sep=\"\\t\")\n",
    "raw_df = raw_df.drop(columns=[\"dbgap_dnaseq_sample\", \"dbgap_rnaseq_sample\", \"run_index\", \"plate_num\", \"time_of_read\", \"passed_qc\"])\n",
    "raw_df = raw_df.drop_duplicates()\n",
    "raw_df.head()"
   ],
   "id": "9b6a50cb999ed772",
   "outputs": [],
   "execution_count": null
  },
  {
   "metadata": {},
   "cell_type": "code",
   "source": [
    "raw_df[\"rounddose\"] = raw_df[\"well_concentration\"].round(10)\n",
    "raw_df[\"mindose\"] = raw_df.groupby([\"dbgap_subject_id\", \"inhibitor\"], as_index=False)[\"rounddose\"].transform(min)\n",
    "raw_df[\"maxdose\"] = raw_df.groupby([\"dbgap_subject_id\", \"inhibitor\"], as_index=False)[\"rounddose\"].transform(max)\n",
    "raw_df[\"repl\"] = raw_df.groupby([\"dbgap_subject_id\", \"inhibitor\", \"rounddose\"], as_index=False)[\"rounddose\"].cumcount()\n",
    "raw_df.head()"
   ],
   "id": "f2f91d95231a7fa0",
   "outputs": [],
   "execution_count": null
  },
  {
   "metadata": {},
   "cell_type": "code",
   "source": [
    "drug_dosages = raw_df[[\"dbgap_subject_id\", \"inhibitor\", \"mindose\", \"maxdose\"]].drop_duplicates()\n",
    "drug_dosages.to_csv(\"drug_dosages_BeatAML2.csv\", index=False)"
   ],
   "id": "3d4cf2d88d4a1d57",
   "outputs": [],
   "execution_count": null
  },
  {
   "metadata": {},
   "cell_type": "markdown",
   "source": "### Drug name to PubChem mapping",
   "id": "813e80af20ffb3ea"
  },
  {
   "metadata": {},
   "cell_type": "code",
   "source": [
    "import pubchempy as pcp\n",
    "results = {}\n",
    "chembls = {}\n",
    "manual_review = {}\n",
    "inchis = {}\n",
    "i = 0\n",
    "for drug in raw_df['inhibitor'].unique():\n",
    "    if i % 10 == 0:\n",
    "        print(i, '/', len(raw_df['inhibitor'].unique()))\n",
    "    compounds = pcp.get_compounds(identifier=drug, namespace=\"name\")\n",
    "    if len(compounds) == 0:\n",
    "        manual_review[drug] = [compounds]\n",
    "    else:\n",
    "        results[drug] = compounds[0].cid\n",
    "        inchis = compounds[0].inchi\n",
    "        chembl = None\n",
    "        for syn in compounds[0].synonyms or []:\n",
    "            if syn.startswith(\"CHEMBL\"):\n",
    "                chembl = syn\n",
    "                break\n",
    "        chembls[drug] = chembl\n",
    "    i += 1"
   ],
   "id": "334e2cb7e1d1c5e3",
   "outputs": [],
   "execution_count": null
  },
  {
   "metadata": {},
   "cell_type": "code",
   "source": [
    "print(f'{len(results)} matches, {len(manual_review)} not found')\n",
    "manual_review |= {brd: cid for brd, cid in results.items() if cid is None}\n",
    "manual_review"
   ],
   "id": "5399e2940633989b",
   "outputs": [],
   "execution_count": null
  },
  {
   "metadata": {},
   "cell_type": "code",
   "source": [
    "manual_review['Pazopanib (GW786034)'] = '10113978' # is Pazopanib\n",
    "manual_review['Roscovitine (CYC-202)'] = '160355' # is Roscovitine\n",
    "manual_review['Afatinib (BIBW-2992)'] = '10184653' # is Afatinib\n",
    "manual_review['Axitinib (AG-013736)'] = '6450551' # is Axitinib\n",
    "manual_review['Foretinib (XL880)'] = '42642645' # is Foretinib\n",
    "manual_review['Lestaurtinib (CEP-701)'] = '126565' # is Lestaurtinib\n",
    "manual_review['Masitinib (AB-1010)'] = '10074640' # is Masitinib\n",
    "manual_review['Vargetef'] = '135423438' # typo (Vargatef) = Nintedanib\n",
    "manual_review['Vemurafenib (PLX-4032)'] = '42611257' # is Vemurafenib\n",
    "manual_review['Volasertib (BI-6727)'] = '10461508' # is Volasertib\n",
    "manual_review['AT-101'] = '3503' # Gossypol; AT 101 is synonym\n",
    "manual_review['Baiclein'] = '5281605' # typo, is Baicalein\n",
    "manual_review['Dovitinib (CHIR-258)'] = '135398510' # was empty"
   ],
   "id": "f4609219c57d67c5",
   "outputs": [],
   "execution_count": null
  },
  {
   "metadata": {},
   "cell_type": "code",
   "source": [
    "def sanitize(val):\n",
    "    if isinstance(val, int):\n",
    "        return str(val)\n",
    "    if isinstance(val, list):\n",
    "        return ','.join([str(i) for i in val])\n",
    "    return val\n",
    "cids = pd.Series(results | manual_review, name=\"CID\").apply(sanitize).astype(str)\n",
    "cids"
   ],
   "id": "14ecdbdc",
   "outputs": [],
   "execution_count": null
  },
  {
   "metadata": {},
   "cell_type": "code",
   "source": [
    "chembls = pd.Series(chembls, name=\"CHEMBL\")\n",
    "raw_df = raw_df.drop(columns=['inhibitor_panel', 'replicate', 'well_concentration'])\n",
    "full_df = raw_df.merge(cids, left_on='inhibitor', right_index=True).merge(chembls, left_on='inhibitor', right_index=True, how='left')\n",
    "# The required columns are \"dose\", \"response\", \"sample\", and \"drug\", with an optional \"replicate\" column.\n",
    "full_df.rename(columns={'dbgap_subject_id': 'sample', 'inhibitor': 'drug_name', 'rounddose': 'dose', 'CID': 'drug', 'normalized_viability': 'response', 'repl': 'replicate'}, inplace=True)\n",
    "full_df.to_csv('viabilities_processed.csv', index=None)\n",
    "full_df"
   ],
   "id": "27439eaebb2805c9",
   "outputs": [],
   "execution_count": null
  },
  {
   "metadata": {},
   "cell_type": "code",
   "source": [
    "# check for duplicates\n",
    "print(len(full_df))\n",
    "print(len(full_df[[\"sample\", \"drug\", \"dose\", \"response\", \"replicate\"]].drop_duplicates()))\n",
    "full_df[full_df[[\"sample\", \"drug\", \"dose\", \"response\", \"replicate\"]].duplicated()]"
   ],
   "id": "1dc7ff1da967fe8f",
   "outputs": [],
   "execution_count": null
  },
  {
   "metadata": {},
   "cell_type": "markdown",
   "source": "## DrEvalPy: run CurveCurator",
   "id": "73c39c82471a0d34"
  },
  {
   "metadata": {},
   "cell_type": "code",
   "source": "preprocess(input_file='viabilities_processed.csv', output_dir='curvecurator', dataset_name='BeatAML2', cores=6, normalize=True)",
   "id": "3211a84d",
   "outputs": [],
   "execution_count": null
  },
  {
   "metadata": {},
   "cell_type": "code",
   "source": "_exec_curvecurator(Path('curvecurator'))",
   "id": "7ec7202772656ca",
   "outputs": [],
   "execution_count": null
  },
  {
   "metadata": {},
   "cell_type": "code",
   "source": "postprocess('curvecurator', dataset_name='BeatAML2')",
   "id": "c776a56e3e91d129",
   "outputs": [],
   "execution_count": null
  },
  {
   "metadata": {},
   "cell_type": "markdown",
   "source": [
    "## Postprocess for Zenodo\n",
    "\n",
    "* Quality control\n",
    "* Set EC50s outside the measured range to NaN\n",
    "* Set IC50s outside 10*max/10**-1 min measured range to NaN\n"
   ],
   "id": "404aae008530c66b"
  },
  {
   "metadata": {},
   "cell_type": "code",
   "source": [
    "viability_df = pd.read_csv('curvecurator/Bruna.csv', converters={'sample': str, 'pubchem_id': str})\n",
    "full_df = pd.read_csv('viabilities_processed.csv', converters={'drug': str})[['sample', 'drug', 'drug_name', 'CHEMBL', 'mindose', 'maxdose', ]].drop_duplicates()\n",
    "viability_df = viability_df.merge(full_df, left_on=['cell_line_name', 'pubchem_id'], right_on=['sample', 'drug'])"
   ],
   "id": "ae024d2cc423705d",
   "outputs": [],
   "execution_count": null
  },
  {
   "metadata": {},
   "cell_type": "code",
   "source": [
    "# join with original data\n",
    "original_measures_df = pd.read_csv('beataml_probit_curve_fits_v4_dbgap.txt', sep='\\t')\n",
    "viability_df = viability_df.merge(original_measures_df, left_on=['cell_line_name', 'drug_name'], right_on=['dbgap_subject_id', 'inhibitor'])\n",
    "viability_df = viability_df.rename(columns={'ic50': 'IC50', 'auc': 'AUC'})"
   ],
   "id": "d7866ff72b3b630",
   "outputs": [],
   "execution_count": null
  },
  {
   "metadata": {},
   "cell_type": "code",
   "source": [
    "import seaborn as sns\n",
    "import matplotlib.pyplot as plt\n",
    "import numpy as np\n",
    "# checks out because they measure AUC=area under the dose response curve, we measure AUC=area under the dose viability curve\n",
    "sns.scatterplot(viability_df, x=\"AUC_curvecurator\", y=\"AUC\", s=1)"
   ],
   "id": "eac68a6395ac09de",
   "outputs": [],
   "execution_count": null
  },
  {
   "metadata": {},
   "cell_type": "code",
   "source": [
    "sns.scatterplot(viability_df, x=\"IC50_curvecurator\", y=\"IC50\", s=1)\n",
    "plt.xlim(1e-4, 1e2)\n",
    "plt.ylim(1e-4, 1e2)\n",
    "plt.xscale('log')\n",
    "plt.yscale('log')"
   ],
   "id": "edd67585f4e74f5f",
   "outputs": [],
   "execution_count": null
  },
  {
   "metadata": {},
   "cell_type": "code",
   "source": "",
   "id": "50007e9e721fdd38",
   "outputs": [],
   "execution_count": null
  },
  {
   "metadata": {},
   "cell_type": "code",
   "source": [
    "# create cell line names file\n",
    "response_data = data.merge(df_map, left_on=\"inhibitor\", right_on=\"drug_name_beataml\", how=\"left\")\n",
    "response_data[\"LN_IC50\"] = response_data[\"ic50\"]\n",
    "response_data.drop(columns=[\"ic50\", \"drug_name_beataml\", \"status\", \"paper_inclusion\"], inplace=True)\n",
    "response_data[\"cell_line_name\"] = response_data[\"dbgap_subject_id\"].astype(str)\n",
    "\n",
    "cell_line_names = response_data.drop_duplicates(\"cell_line_name\")[[\"cell_line_name\"]]\n",
    "cell_line_names[\"Cellosaurus_id\"] = pd.NA\n",
    "cell_line_names[\"tissue\"] = \"Blood\"\n",
    "cell_line_names.to_csv(\"cell_line_names.csv\", index=False)"
   ],
   "id": "e94c504f",
   "outputs": [],
   "execution_count": null
  },
  {
   "metadata": {},
   "cell_type": "code",
   "source": [
    "# create gene expression feature file\n",
    "gene_exp = pd.read_csv(\"other/beataml_waves1to4_norm_exp_dbgap.txt\", sep=\"\\t\", index_col=0)\n",
    "\n",
    "gene_exp_processed = gene_exp\n",
    "gene_exp_processed =  gene_exp_processed.reset_index().drop(columns=[\"stable_id\", \"description\", \"biotype\"]).set_index(\"display_label\")\n",
    "gene_exp_processed = gene_exp_processed.T\n",
    "map_rna_id_to_cl_name = {rna_ind: sub_ind for sub_ind, rna_ind in zip(response_data[\"cell_line_name\"], response_data[\"dbgap_rnaseq_sample\"])}\n",
    "#remap index\n",
    "gene_exp_processed.index = [map_rna_id_to_cl_name.get(ind, pd.NA) for ind in gene_exp_processed.index]\n",
    "gene_exp_processed =  gene_exp_processed.drop(pd.NA)\n",
    "gene_exp_processed.to_csv(\"gene_expression.csv\", index=True)"
   ],
   "id": "660d5da0",
   "outputs": [],
   "execution_count": null
  },
  {
   "metadata": {},
   "cell_type": "code",
   "source": [
    "# create drug names file\n",
    "drug_names = df_map[[\"drug_name_beataml\", \"drug_name\", \"pubchem_id\"]]\n",
    "#if pubchemid is NA, use drug_name\n",
    "drug_names[\"pubchem_id\"] = drug_names[\"pubchem_id\"].replace(\"\", pd.NA)\n",
    "drug_names[\"pubchem_id\"] = drug_names[\"pubchem_id\"].replace(\" \", pd.NA)\n",
    "\n",
    "drug_names[\"pubchem_id\"] = drug_names[\"pubchem_id\"].fillna(drug_names[\"drug_name\"])\n",
    "\n",
    "drug_names.to_csv(\"drug_names.csv\", index=False)"
   ],
   "id": "89e9d7fb",
   "outputs": [],
   "execution_count": null
  },
  {
   "metadata": {},
   "cell_type": "code",
   "source": [
    "#create fingerprints\n",
    "import os\n",
    "import numpy as np\n",
    "import pandas as pd\n",
    "from rdkit import Chem\n",
    "from rdkit.Chem import AllChem, DataStructs\n",
    "\n",
    "# df has columns: pubchem_id, canonical_smiles\n",
    "drug_smiles = pd.read_csv(\"drug_smiles.csv\")\n",
    "\n",
    "df = drug_smiles.copy()\n",
    "\n",
    "out_dir = \"drug_fingerprints\"\n",
    "os.makedirs(out_dir, exist_ok=True)\n",
    "\n",
    "def morgan_bits(smiles, n_bits, radius=2):\n",
    "    m = Chem.MolFromSmiles(smiles)\n",
    "    if m is None:\n",
    "        return np.full(n_bits, np.nan)  # keep shape even if invalid\n",
    "    fp = AllChem.GetMorganFingerprintAsBitVect(m, radius, nBits=n_bits)\n",
    "    arr = np.zeros((n_bits,), dtype=int)\n",
    "    DataStructs.ConvertToNumpyArray(fp, arr)\n",
    "    return arr\n",
    "\n",
    "sizes = [64, 128, 256, 512, 1024, 2048]\n",
    "\n",
    "for n in sizes:\n",
    "    print(f\"computing {n}-bit fingerprints\")\n",
    "    # compute and stack into a DataFrame (rows = bit positions, cols = pubchem_id)\n",
    "    fp_matrix = {}\n",
    "    for pid, smi in zip(df[\"pubchem_id\"], df[\"canonical_smiles\"]):\n",
    "        fp_matrix[pid] = morgan_bits(smi, n)\n",
    "    fp_df = pd.DataFrame(fp_matrix)  # index 0..n-1, columns = pubchem_id\n",
    "    fp_df.to_csv(f\"{out_dir}/pubchem_id_to_demorgan_{n}_map.csv\", index=False)\n"
   ],
   "id": "6bdc2aae",
   "outputs": [],
   "execution_count": null
  },
  {
   "metadata": {},
   "cell_type": "code",
   "source": [
    "# rename genes in gene expression file  to correct symbols\n",
    "ge = pd.read_csv(\"gene_expression.csv\", index_col=0)\n",
    "genes = list(ge.columns)\n",
    "new_gene_names = {\n",
    "    \"AARS\": \"AARS1\",\n",
    "    \"EPRS\": \"EPRS1\",\n",
    "    \"FAM57A\": \"TLCD3A\",\n",
    "    \"FAM69A\": \"DIPK1A\",\n",
    "    \"H2AFV\": \"H2AZ2\",\n",
    "    \"HIST1H2BK\": \"H2BC12\",\n",
    "    \"HIST2H2BE\": \"H2BC21\",\n",
    "    \"KIAA0100\": \"BLTP2\",\n",
    "    \"KIAA0355\": \"GARRE1\",\n",
    "    \"NARFL\": \"CIAO3\",\n",
    "    \"PAPD7\": \"TENT4A\",\n",
    "    \"SKIV2L\": \"SKIC2\",\n",
    "    \"TSTA3\": \"GFUS\",\n",
    "    \"WDR61\": \"SKIC8\",\n",
    "    \"WRB\": \"GET1\",\n",
    "}\n",
    "new_genes = [new_gene_names.get(g, g) for g in genes]\n",
    "ge.columns = new_genes\n",
    "landmark_genes  = pd.read_csv(\"gene_lists/landmark_genes.csv\", index_col=0)\n",
    "set(landmark_genes[\"Symbol\"]) - set(new_genes)\n",
    "#save again\n",
    "ge.index.name = \"cell_line_name\"\n",
    "ge.to_csv(\"gene_expression.csv\", index=True)"
   ],
   "id": "eff0e075",
   "outputs": [],
   "execution_count": null
  },
  {
   "metadata": {},
   "cell_type": "code",
   "source": [
    "# correct pubchem ids in BeatAML2.csv\n",
    "import pandas as pd\n",
    "\n",
    "d = pd.read_csv(\"BeatAML2.csv\")\n",
    "d[\"pubchem_id\"] = d[\"pubchem_id\"].astype(str).str.replace(\".0\", \"\", regex=False)\n",
    "\n",
    "d.loc[d[\"drug_name\"] == \"MEK1/2 Inhibitor\", \"pubchem_id\"] = \"MEK1/2 Inhibitor\"\n",
    "d.loc[d[\"drug_name\"] == \"JQ12\", \"pubchem_id\"] = \"JQ12\"\n",
    "\n",
    "d.to_csv(\"BeatAML2.csv\", index=False)\n"
   ],
   "id": "21c005a6",
   "outputs": [],
   "execution_count": null
  },
  {
   "metadata": {},
   "cell_type": "code",
   "source": "",
   "id": "5a6863f10d4528d7",
   "outputs": [],
   "execution_count": null
  }
 ],
 "metadata": {
  "kernelspec": {
   "display_name": "drp2",
   "language": "python",
   "name": "python3"
  },
  "language_info": {
   "codemirror_mode": {
    "name": "ipython",
    "version": 3
   },
   "file_extension": ".py",
   "mimetype": "text/x-python",
   "name": "python",
   "nbconvert_exporter": "python",
   "pygments_lexer": "ipython3",
   "version": "3.13.3"
  }
 },
 "nbformat": 4,
 "nbformat_minor": 5
}
