{
 "cells": [
  {
   "cell_type": "code",
   "id": "initial_id",
   "metadata": {
    "collapsed": true,
    "ExecuteTime": {
     "end_time": "2025-11-13T09:24:01.531189Z",
     "start_time": "2025-11-13T09:24:01.512978Z"
    }
   },
   "source": [
    "import pandas as pd\n",
    "from pathlib import Path"
   ],
   "outputs": [],
   "execution_count": 13
  },
  {
   "metadata": {},
   "cell_type": "markdown",
   "source": "Load all datasets for intersection lists",
   "id": "dc30a00a04b40c13"
  },
  {
   "metadata": {
    "ExecuteTime": {
     "end_time": "2025-11-13T09:24:02.724966Z",
     "start_time": "2025-11-13T09:24:02.721129Z"
    }
   },
   "cell_type": "code",
   "source": [
    "all_datasets= ['BeatAML2', 'CCLE', 'CTRPv1', 'CTRPv2', 'GDSC1', 'GDSC2', 'PDX_Bruna']\n",
    "path_to_data = '../../drevalpy/data'"
   ],
   "id": "214633b53bca938d",
   "outputs": [],
   "execution_count": 14
  },
  {
   "metadata": {
    "ExecuteTime": {
     "end_time": "2025-11-13T09:24:03.389235Z",
     "start_time": "2025-11-13T09:24:03.375714Z"
    }
   },
   "cell_type": "code",
   "source": [
    "def get_gene_intersection_omic(dataset_list: list, data_path: str, omic: str):\n",
    "    first_dataset = dataset_list.pop()\n",
    "    print(f'Processing {first_dataset}...')\n",
    "    first_path = Path(data_path, first_dataset, f'{omic}.csv')\n",
    "    df = pd.read_csv(first_path, index_col='cell_line_name')\n",
    "    if 'cellosaurus_id' in df.columns:\n",
    "        df.drop(columns=['cellosaurus_id'], inplace=True)\n",
    "    gene_set = set(df.columns)\n",
    "    for ds in dataset_list:\n",
    "        print(f'Processing {ds}...')\n",
    "        path = Path(data_path, ds, f'{omic}.csv')\n",
    "        df = pd.read_csv(path, index_col='cell_line_name')\n",
    "        if 'cellosaurus_id' in df.columns:\n",
    "            df.drop(columns=['cellosaurus_id'], inplace=True)\n",
    "        gene_set = gene_set.intersection(set(df.columns))\n",
    "    return gene_set"
   ],
   "id": "ae487b881f74c523",
   "outputs": [],
   "execution_count": 15
  },
  {
   "metadata": {
    "ExecuteTime": {
     "end_time": "2025-11-13T09:24:20.518927Z",
     "start_time": "2025-11-13T09:24:03.956124Z"
    }
   },
   "cell_type": "code",
   "source": [
    "gene_set_cnv = get_gene_intersection_omic(\n",
    "    dataset_list=['CCLE', 'CTRPv1', 'CTRPv2', 'GDSC1', 'GDSC2', 'PDX_Bruna'],\n",
    "    data_path=path_to_data,\n",
    "    omic='copy_number_variation_gistic'\n",
    ")\n",
    "pd.DataFrame({'Symbol': list(gene_set_cnv)}).to_csv('copy_number_variation_gistic_intersection.csv')"
   ],
   "id": "67a20d3d58800ed3",
   "outputs": [
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "Processing PDX_Bruna...\n",
      "Processing CCLE...\n",
      "Processing CTRPv1...\n",
      "Processing CTRPv2...\n",
      "Processing GDSC1...\n",
      "Processing GDSC2...\n"
     ]
    }
   ],
   "execution_count": 16
  },
  {
   "metadata": {
    "ExecuteTime": {
     "end_time": "2025-11-13T09:25:08.359237Z",
     "start_time": "2025-11-13T09:24:20.550389Z"
    }
   },
   "cell_type": "code",
   "source": [
    "gene_set_gex = get_gene_intersection_omic(\n",
    "    dataset_list=all_datasets,\n",
    "    data_path=path_to_data,\n",
    "    omic='gene_expression'\n",
    ")\n",
    "pd.DataFrame({'Symbol': list(gene_set_gex)}).to_csv('gene_expression_intersection.csv')"
   ],
   "id": "f4a90a9a28013949",
   "outputs": [
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "Processing PDX_Bruna...\n",
      "Processing BeatAML2...\n",
      "Processing CCLE...\n",
      "Processing CTRPv1...\n",
      "Processing CTRPv2...\n",
      "Processing GDSC1...\n",
      "Processing GDSC2...\n"
     ]
    }
   ],
   "execution_count": 17
  },
  {
   "metadata": {
    "ExecuteTime": {
     "end_time": "2025-11-13T09:25:15.285681Z",
     "start_time": "2025-11-13T09:25:08.396533Z"
    }
   },
   "cell_type": "code",
   "source": [
    "gene_set_met = get_gene_intersection_omic(\n",
    "    dataset_list=['CCLE', 'CTRPv1', 'CTRPv2', 'GDSC1', 'GDSC2'],\n",
    "    data_path=path_to_data,\n",
    "    omic='methylation'\n",
    ")\n",
    "pd.DataFrame({'Symbol': list(gene_set_met)}).to_csv('methylation_intersection.csv')"
   ],
   "id": "914e858f7a76f866",
   "outputs": [
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "Processing GDSC2...\n",
      "Processing CCLE...\n",
      "Processing CTRPv1...\n",
      "Processing CTRPv2...\n",
      "Processing GDSC1...\n"
     ]
    }
   ],
   "execution_count": 18
  },
  {
   "metadata": {
    "ExecuteTime": {
     "end_time": "2025-11-13T09:35:04.048255Z",
     "start_time": "2025-11-13T09:34:45.787865Z"
    }
   },
   "cell_type": "code",
   "source": [
    "gene_set_mut = get_gene_intersection_omic(\n",
    "    dataset_list=['CCLE', 'CTRPv1', 'CTRPv2', 'GDSC1', 'GDSC2'],\n",
    "    data_path=path_to_data,\n",
    "    omic='mutations'\n",
    ")\n",
    "pd.DataFrame({'Symbol': list(gene_set_mut)}).to_csv('mutations_intersection.csv')"
   ],
   "id": "35d4f1b027d66046",
   "outputs": [
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "Processing GDSC2...\n",
      "Processing CCLE...\n",
      "Processing CTRPv1...\n",
      "Processing CTRPv2...\n",
      "Processing GDSC1...\n"
     ]
    }
   ],
   "execution_count": 22
  },
  {
   "metadata": {
    "ExecuteTime": {
     "end_time": "2025-11-13T09:47:35.152105Z",
     "start_time": "2025-11-13T09:47:29.516387Z"
    }
   },
   "cell_type": "code",
   "source": [
    "# same for all for now; could change at a later point in time to include the other dataset\n",
    "proteomics = pd.read_csv(Path(path_to_data, 'CCLE', 'proteomics.csv'), index_col=['cellosaurus_id', 'cell_line_name'])\n",
    "pd.DataFrame({'Symbol': list(proteomics.columns)}).to_csv('proteomics_intersection.csv')"
   ],
   "id": "fe4f1dead03551ac",
   "outputs": [],
   "execution_count": 26
  },
  {
   "metadata": {},
   "cell_type": "markdown",
   "source": "## Reduced gene lists",
   "id": "a184cfcc047fd22f"
  },
  {
   "metadata": {},
   "cell_type": "markdown",
   "source": "Todo: how did we get here?",
   "id": "20efe04e54b0ed22"
  },
  {
   "metadata": {
    "ExecuteTime": {
     "end_time": "2025-11-13T09:54:00.624319Z",
     "start_time": "2025-11-13T09:54:00.476059Z"
    }
   },
   "cell_type": "code",
   "source": [
    "landmark_genes = pd.read_csv(\"landmark_genes.csv\")\n",
    "drug_targets = pd.read_csv(\"drug_target_genes_all_drugs.csv\")\n",
    "paccman_list = pd.read_csv(\"gene_list_paccmann_network_prop.csv\")"
   ],
   "id": "e1cfa572babc64fb",
   "outputs": [],
   "execution_count": 28
  },
  {
   "metadata": {
    "ExecuteTime": {
     "end_time": "2025-11-13T09:56:49.227101Z",
     "start_time": "2025-11-13T09:56:49.179645Z"
    }
   },
   "cell_type": "code",
   "source": [
    "# proteomics versions\n",
    "landmark_proteomics = set(landmark_genes['Symbol']).intersection(set(proteomics.columns))\n",
    "drug_targets_proteomics = set(drug_targets['Symbol']).intersection(set(proteomics.columns))\n",
    "paccman_list_proteomics = set(paccman_list['Symbol']).intersection(set(proteomics.columns))"
   ],
   "id": "5a441aa6244d0c1c",
   "outputs": [],
   "execution_count": 29
  },
  {
   "metadata": {
    "ExecuteTime": {
     "end_time": "2025-11-13T09:59:36.947641Z",
     "start_time": "2025-11-13T09:59:36.920425Z"
    }
   },
   "cell_type": "code",
   "source": [
    "pd.DataFrame({'Symbol': list(landmark_proteomics)}).to_csv('landmark_genes_proteomics.csv')\n",
    "pd.DataFrame({'Symbol': list(drug_targets_proteomics)}).to_csv('drug_target_genes_all_drugs_proteomics.csv')\n",
    "pd.DataFrame({'Symbol': list(paccman_list_proteomics)}).to_csv('gene_list_paccmann_network_prop_proteomics.csv')"
   ],
   "id": "4b3f9fb26ca0a4bd",
   "outputs": [],
   "execution_count": 34
  },
  {
   "metadata": {
    "ExecuteTime": {
     "end_time": "2025-11-13T10:01:39.905959Z",
     "start_time": "2025-11-13T10:01:39.848084Z"
    }
   },
   "cell_type": "code",
   "source": "all_genes_intersected = gene_set_cnv.intersection(gene_set_gex).intersection(gene_set_mut).intersection(set(proteomics.columns))",
   "id": "cb5542591cd2bc2e",
   "outputs": [],
   "execution_count": 35
  },
  {
   "metadata": {
    "ExecuteTime": {
     "end_time": "2025-11-13T10:02:38.778905Z",
     "start_time": "2025-11-13T10:02:38.713820Z"
    }
   },
   "cell_type": "code",
   "source": [
    "landmark_reduced = set(landmark_genes['Symbol']).intersection(all_genes_intersected)\n",
    "drug_targets_reduced = set(drug_targets['Symbol']).intersection(all_genes_intersected)\n",
    "paccman_list_reduced = set(paccman_list['Symbol']).intersection(all_genes_intersected)"
   ],
   "id": "b85fbb4245e98392",
   "outputs": [],
   "execution_count": 36
  },
  {
   "metadata": {
    "ExecuteTime": {
     "end_time": "2025-11-13T10:04:04.585050Z",
     "start_time": "2025-11-13T10:04:04.550701Z"
    }
   },
   "cell_type": "code",
   "source": [
    "pd.DataFrame({'Symbol': list(landmark_reduced)}).to_csv('landmark_genes_reduced.csv')\n",
    "pd.DataFrame({'Symbol': list(drug_targets_reduced)}).to_csv('drug_target_genes_reduced.csv')\n",
    "pd.DataFrame({'Symbol': list(paccman_list_reduced)}).to_csv('gene_list_paccmann_network_prop_reduced.csv')"
   ],
   "id": "fb3f8a2f52ca1b5",
   "outputs": [],
   "execution_count": 37
  },
  {
   "metadata": {},
   "cell_type": "code",
   "outputs": [],
   "execution_count": null,
   "source": "",
   "id": "d62e4b703fb345e4"
  }
 ],
 "metadata": {
  "kernelspec": {
   "display_name": "Python 3",
   "language": "python",
   "name": "python3"
  },
  "language_info": {
   "codemirror_mode": {
    "name": "ipython",
    "version": 2
   },
   "file_extension": ".py",
   "mimetype": "text/x-python",
   "name": "python",
   "nbconvert_exporter": "python",
   "pygments_lexer": "ipython2",
   "version": "2.7.6"
  }
 },
 "nbformat": 4,
 "nbformat_minor": 5
}
