{
 "cells": [
  {
   "cell_type": "code",
   "id": "initial_id",
   "metadata": {
    "collapsed": true,
    "ExecuteTime": {
     "end_time": "2025-11-12T17:44:38.674202Z",
     "start_time": "2025-11-12T17:44:38.611810Z"
    }
   },
   "source": "import pandas as pd",
   "outputs": [],
   "execution_count": 51
  },
  {
   "metadata": {
    "ExecuteTime": {
     "end_time": "2025-11-12T17:52:16.562799Z",
     "start_time": "2025-11-12T17:52:16.114587Z"
    }
   },
   "cell_type": "code",
   "source": [
    "# read in response for comparisons\n",
    "response = pd.read_csv('response/BeatAML2.csv', converters={'cell_line_name': str})"
   ],
   "id": "fdac761bfd901e75",
   "outputs": [],
   "execution_count": 62
  },
  {
   "metadata": {},
   "cell_type": "markdown",
   "source": "## Gene expression",
   "id": "f19192e3c1918408"
  },
  {
   "metadata": {
    "ExecuteTime": {
     "end_time": "2025-11-12T15:15:09.213919Z",
     "start_time": "2025-11-12T15:15:07.581067Z"
    }
   },
   "cell_type": "code",
   "source": [
    "# create gene expression feature file\n",
    "gene_exp = pd.read_csv(\"gene_expression/beataml_waves1to4_norm_exp_dbgap.txt\", sep=\"\\t\", index_col=0)\n",
    "gene_exp = gene_exp.reset_index().drop(columns=[\"stable_id\", \"description\", \"biotype\"]).set_index(\"display_label\")\n",
    "gene_exp_processed = gene_exp.T\n",
    "gene_exp_processed = gene_exp_processed.reset_index().rename(columns={\"index\": \"dbgap_rnaseq_sample\"})\n",
    "gene_exp_processed.iloc[0:5, 0:5]"
   ],
   "id": "ec1a6b9b91182fc3",
   "outputs": [
    {
     "data": {
      "text/plain": [
       "display_label dbgap_rnaseq_sample    TSPAN6      DPM1     SCYL3  C1orf112\n",
       "0                         BA2392R  1.523670  7.107711  3.362605  3.881649\n",
       "1                         BA2611R  1.856566  6.865282  3.320940  4.045935\n",
       "2                         BA2506R  1.701307  6.889932  3.407834  3.904311\n",
       "3                         BA2430R  1.870627  6.969120  3.265446  4.057645\n",
       "4                         BA2448R  1.699496  6.884921  3.428475  3.931659"
      ],
      "text/html": [
       "<div>\n",
       "<style scoped>\n",
       "    .dataframe tbody tr th:only-of-type {\n",
       "        vertical-align: middle;\n",
       "    }\n",
       "\n",
       "    .dataframe tbody tr th {\n",
       "        vertical-align: top;\n",
       "    }\n",
       "\n",
       "    .dataframe thead th {\n",
       "        text-align: right;\n",
       "    }\n",
       "</style>\n",
       "<table border=\"1\" class=\"dataframe\">\n",
       "  <thead>\n",
       "    <tr style=\"text-align: right;\">\n",
       "      <th>display_label</th>\n",
       "      <th>dbgap_rnaseq_sample</th>\n",
       "      <th>TSPAN6</th>\n",
       "      <th>DPM1</th>\n",
       "      <th>SCYL3</th>\n",
       "      <th>C1orf112</th>\n",
       "    </tr>\n",
       "  </thead>\n",
       "  <tbody>\n",
       "    <tr>\n",
       "      <th>0</th>\n",
       "      <td>BA2392R</td>\n",
       "      <td>1.523670</td>\n",
       "      <td>7.107711</td>\n",
       "      <td>3.362605</td>\n",
       "      <td>3.881649</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>1</th>\n",
       "      <td>BA2611R</td>\n",
       "      <td>1.856566</td>\n",
       "      <td>6.865282</td>\n",
       "      <td>3.320940</td>\n",
       "      <td>4.045935</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>2</th>\n",
       "      <td>BA2506R</td>\n",
       "      <td>1.701307</td>\n",
       "      <td>6.889932</td>\n",
       "      <td>3.407834</td>\n",
       "      <td>3.904311</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>3</th>\n",
       "      <td>BA2430R</td>\n",
       "      <td>1.870627</td>\n",
       "      <td>6.969120</td>\n",
       "      <td>3.265446</td>\n",
       "      <td>4.057645</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>4</th>\n",
       "      <td>BA2448R</td>\n",
       "      <td>1.699496</td>\n",
       "      <td>6.884921</td>\n",
       "      <td>3.428475</td>\n",
       "      <td>3.931659</td>\n",
       "    </tr>\n",
       "  </tbody>\n",
       "</table>\n",
       "</div>"
      ]
     },
     "execution_count": 3,
     "metadata": {},
     "output_type": "execute_result"
    }
   ],
   "execution_count": 3
  },
  {
   "metadata": {
    "ExecuteTime": {
     "end_time": "2025-11-12T15:15:09.511413Z",
     "start_time": "2025-11-12T15:15:09.297176Z"
    }
   },
   "cell_type": "code",
   "source": [
    "mapping = pd.read_excel(\"beataml_waves1to4_sample_mapping.xlsx\", sheet_name='sample_map')\n",
    "mapping.head()"
   ],
   "id": "dbf76a2979f9cee2",
   "outputs": [
    {
     "data": {
      "text/plain": [
       "   patientId  dbgap_subject_id     labId dbgap_rnaseq_sample rna_seq_id  \\\n",
       "0          0              2096  00-00002             BA2392R   00-00002   \n",
       "1          0              2096  00-00003             BA2611R   00-00003   \n",
       "2          0              2096  00-00004             BA2506R   00-00004   \n",
       "3          0              2096  00-00005             BA2430R   00-00005   \n",
       "4          0              2096  00-00006             BA2448R   00-00006   \n",
       "\n",
       "            rna_control rna_include_in_analysis dbgap_dnaseq_sample  \\\n",
       "0  Healthy pooled CD34+                     yes                 NaN   \n",
       "1  Healthy pooled CD34+                     yes                 NaN   \n",
       "2  Healthy pooled CD34+                     yes                 NaN   \n",
       "3  Healthy pooled CD34+                     yes                 NaN   \n",
       "4  Healthy pooled CD34+                     yes                 NaN   \n",
       "\n",
       "  dna_seq_id dna_capture_type dna_include_in_analysis analysisDrug  \n",
       "0        NaN              NaN                     NaN          NaN  \n",
       "1        NaN              NaN                     NaN          NaN  \n",
       "2        NaN              NaN                     NaN          NaN  \n",
       "3        NaN              NaN                     NaN          NaN  \n",
       "4        NaN              NaN                     NaN          NaN  "
      ],
      "text/html": [
       "<div>\n",
       "<style scoped>\n",
       "    .dataframe tbody tr th:only-of-type {\n",
       "        vertical-align: middle;\n",
       "    }\n",
       "\n",
       "    .dataframe tbody tr th {\n",
       "        vertical-align: top;\n",
       "    }\n",
       "\n",
       "    .dataframe thead th {\n",
       "        text-align: right;\n",
       "    }\n",
       "</style>\n",
       "<table border=\"1\" class=\"dataframe\">\n",
       "  <thead>\n",
       "    <tr style=\"text-align: right;\">\n",
       "      <th></th>\n",
       "      <th>patientId</th>\n",
       "      <th>dbgap_subject_id</th>\n",
       "      <th>labId</th>\n",
       "      <th>dbgap_rnaseq_sample</th>\n",
       "      <th>rna_seq_id</th>\n",
       "      <th>rna_control</th>\n",
       "      <th>rna_include_in_analysis</th>\n",
       "      <th>dbgap_dnaseq_sample</th>\n",
       "      <th>dna_seq_id</th>\n",
       "      <th>dna_capture_type</th>\n",
       "      <th>dna_include_in_analysis</th>\n",
       "      <th>analysisDrug</th>\n",
       "    </tr>\n",
       "  </thead>\n",
       "  <tbody>\n",
       "    <tr>\n",
       "      <th>0</th>\n",
       "      <td>0</td>\n",
       "      <td>2096</td>\n",
       "      <td>00-00002</td>\n",
       "      <td>BA2392R</td>\n",
       "      <td>00-00002</td>\n",
       "      <td>Healthy pooled CD34+</td>\n",
       "      <td>yes</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>1</th>\n",
       "      <td>0</td>\n",
       "      <td>2096</td>\n",
       "      <td>00-00003</td>\n",
       "      <td>BA2611R</td>\n",
       "      <td>00-00003</td>\n",
       "      <td>Healthy pooled CD34+</td>\n",
       "      <td>yes</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>2</th>\n",
       "      <td>0</td>\n",
       "      <td>2096</td>\n",
       "      <td>00-00004</td>\n",
       "      <td>BA2506R</td>\n",
       "      <td>00-00004</td>\n",
       "      <td>Healthy pooled CD34+</td>\n",
       "      <td>yes</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>3</th>\n",
       "      <td>0</td>\n",
       "      <td>2096</td>\n",
       "      <td>00-00005</td>\n",
       "      <td>BA2430R</td>\n",
       "      <td>00-00005</td>\n",
       "      <td>Healthy pooled CD34+</td>\n",
       "      <td>yes</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>4</th>\n",
       "      <td>0</td>\n",
       "      <td>2096</td>\n",
       "      <td>00-00006</td>\n",
       "      <td>BA2448R</td>\n",
       "      <td>00-00006</td>\n",
       "      <td>Healthy pooled CD34+</td>\n",
       "      <td>yes</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "    </tr>\n",
       "  </tbody>\n",
       "</table>\n",
       "</div>"
      ]
     },
     "execution_count": 4,
     "metadata": {},
     "output_type": "execute_result"
    }
   ],
   "execution_count": 4
  },
  {
   "metadata": {
    "ExecuteTime": {
     "end_time": "2025-11-12T15:15:09.744102Z",
     "start_time": "2025-11-12T15:15:09.633322Z"
    }
   },
   "cell_type": "code",
   "source": [
    "gene_exp_processed = gene_exp_processed.merge(mapping[['dbgap_subject_id', 'dbgap_rnaseq_sample']], on='dbgap_rnaseq_sample', how='left')\n",
    "gene_exp_processed = gene_exp_processed.rename(columns={'dbgap_subject_id': 'cell_line_name'})\n",
    "gene_exp_processed = gene_exp_processed.drop(columns=['dbgap_rnaseq_sample'])\n",
    "gene_exp_processed['cell_line_name'] = gene_exp_processed['cell_line_name'].astype(str)\n",
    "gene_exp_processed = gene_exp_processed.set_index('cell_line_name')\n",
    "gene_exp_processed.iloc[0:5, 0:5]"
   ],
   "id": "3ae2cdc28c49bda5",
   "outputs": [
    {
     "data": {
      "text/plain": [
       "                  TSPAN6      DPM1     SCYL3  C1orf112       FGR\n",
       "cell_line_name                                                  \n",
       "2096            1.523670  7.107711  3.362605  3.881649  4.571577\n",
       "2096            1.856566  6.865282  3.320940  4.045935  4.433159\n",
       "2096            1.701307  6.889932  3.407834  3.904311  4.576913\n",
       "2096            1.870627  6.969120  3.265446  4.057645  4.468458\n",
       "2096            1.699496  6.884921  3.428475  3.931659  4.650867"
      ],
      "text/html": [
       "<div>\n",
       "<style scoped>\n",
       "    .dataframe tbody tr th:only-of-type {\n",
       "        vertical-align: middle;\n",
       "    }\n",
       "\n",
       "    .dataframe tbody tr th {\n",
       "        vertical-align: top;\n",
       "    }\n",
       "\n",
       "    .dataframe thead th {\n",
       "        text-align: right;\n",
       "    }\n",
       "</style>\n",
       "<table border=\"1\" class=\"dataframe\">\n",
       "  <thead>\n",
       "    <tr style=\"text-align: right;\">\n",
       "      <th></th>\n",
       "      <th>TSPAN6</th>\n",
       "      <th>DPM1</th>\n",
       "      <th>SCYL3</th>\n",
       "      <th>C1orf112</th>\n",
       "      <th>FGR</th>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>cell_line_name</th>\n",
       "      <th></th>\n",
       "      <th></th>\n",
       "      <th></th>\n",
       "      <th></th>\n",
       "      <th></th>\n",
       "    </tr>\n",
       "  </thead>\n",
       "  <tbody>\n",
       "    <tr>\n",
       "      <th>2096</th>\n",
       "      <td>1.523670</td>\n",
       "      <td>7.107711</td>\n",
       "      <td>3.362605</td>\n",
       "      <td>3.881649</td>\n",
       "      <td>4.571577</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>2096</th>\n",
       "      <td>1.856566</td>\n",
       "      <td>6.865282</td>\n",
       "      <td>3.320940</td>\n",
       "      <td>4.045935</td>\n",
       "      <td>4.433159</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>2096</th>\n",
       "      <td>1.701307</td>\n",
       "      <td>6.889932</td>\n",
       "      <td>3.407834</td>\n",
       "      <td>3.904311</td>\n",
       "      <td>4.576913</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>2096</th>\n",
       "      <td>1.870627</td>\n",
       "      <td>6.969120</td>\n",
       "      <td>3.265446</td>\n",
       "      <td>4.057645</td>\n",
       "      <td>4.468458</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>2096</th>\n",
       "      <td>1.699496</td>\n",
       "      <td>6.884921</td>\n",
       "      <td>3.428475</td>\n",
       "      <td>3.931659</td>\n",
       "      <td>4.650867</td>\n",
       "    </tr>\n",
       "  </tbody>\n",
       "</table>\n",
       "</div>"
      ]
     },
     "execution_count": 5,
     "metadata": {},
     "output_type": "execute_result"
    }
   ],
   "execution_count": 5
  },
  {
   "metadata": {
    "ExecuteTime": {
     "end_time": "2025-11-12T15:15:09.926008Z",
     "start_time": "2025-11-12T15:15:09.917313Z"
    }
   },
   "cell_type": "code",
   "source": [
    "unique_response_samples = set(response['cell_line_name'])\n",
    "unique_gex_samples = set(gene_exp_processed.index)\n",
    "print(f\"Gene expression data missing for {len(unique_response_samples - unique_gex_samples)}/{(len(unique_gex_samples))} response samples\")\n",
    "print(f\"Response data missing for {len(unique_gex_samples - unique_response_samples)}/{(len(unique_response_samples))} response samples\")"
   ],
   "id": "1b1a851b2b033559",
   "outputs": [
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "Gene expression data missing for 82/633 response samples\n",
      "Response data missing for 146/569 response samples\n"
     ]
    }
   ],
   "execution_count": 6
  },
  {
   "metadata": {
    "ExecuteTime": {
     "end_time": "2025-11-12T15:15:10.320187Z",
     "start_time": "2025-11-12T15:15:10.277370Z"
    }
   },
   "cell_type": "code",
   "source": [
    "# filter to retain only cell_line_names in intersection of unique_response_samples, unique_gex_samples\n",
    "gene_exp_processed = gene_exp_processed.loc[list(unique_response_samples & unique_gex_samples)]"
   ],
   "id": "253108f561d71851",
   "outputs": [],
   "execution_count": 7
  },
  {
   "metadata": {
    "ExecuteTime": {
     "end_time": "2025-11-12T15:15:13.680088Z",
     "start_time": "2025-11-12T15:15:13.651138Z"
    }
   },
   "cell_type": "code",
   "source": [
    "# check genes\n",
    "gene_list = pd.read_csv('../gene_lists/landmark_genes.csv', index_col=0)\n",
    "new_gene_names = {\n",
    "    \"AARS\": \"AARS1\",\n",
    "    \"SPATA5L1\": \"AFG2B\",\n",
    "    \"KIAA0100\": \"BLTP2\",\n",
    "    \"UHRF1BP1L\": \"BLTP3B\",\n",
    "    \"KIAA2026\": \"BRD10\",\n",
    "    \"BACH1\": \"BRIP1\",\n",
    "    \"NARFL\": \"CIAO3\",\n",
    "    \"CLECL1\": \"CLECL1P\",\n",
    "    \"FAM69A\": \"DIPK1A\",\n",
    "    \"TCTEX1D4\": \"DYNLT4\",\n",
    "    \"FAM102B\": \"EEIG2\",\n",
    "    \"EPRS\": \"EPRS1\",\n",
    "    \"ALS2CR12\": \"FLACC1\",\n",
    "    \"KIAA0355\": \"GARRE1\",\n",
    "    \"WRB\": \"GET1\",\n",
    "    \"TSTA3\": \"GFUS\",\n",
    "    \"H2AFV\": \"H2AZ2\",\n",
    "    \"HIST1H2BK\": \"H2BC12\",\n",
    "    \"HIST2H2BE\": \"H2BC21\",\n",
    "    \"IARS\": \"IARS1\",\n",
    "    \"SSFA2\": \"ITPRID2\",\n",
    "    \"KIAA0556\": \"KATNIP\",\n",
    "    \"HK2\": \"KLK2\",\n",
    "    \"MARCH3\": \"MARCHF3\",\n",
    "    \"CLGN\": \"MMP1\",\n",
    "    \"MAL\": \"MRTFA\",\n",
    "    \"ARMC4\": \"ODAD2\",\n",
    "    \"QARS\": \"QARS1\",\n",
    "    \"SKIV2L\": \"SKIC2\",\n",
    "    \"WDR61\": \"SKIC8\",\n",
    "    \"DIRC2\": \"SLC49A4\",\n",
    "    \"FAM19A2\": \"TAFA2\",\n",
    "    \"TARSL2\": \"TARS3\",\n",
    "    \"PAPD7\": \"TENT4A\",\n",
    "    \"FAM57A\": \"TLCD3A\",\n",
    "    \"FAM57B\": \"TLCD3B\",\n",
    "    \"VNN3\": \"VNN3P\",\n",
    "    \"WARS\": \"WARS1\",\n",
    "}\n",
    "unmapped_genes = {\"ALK\", \"BRDT\", \"ERBB4\", \"EDNRA\", \"EPHA3\", \"GRB7\", \"NR1H4\", \"PRR15L\", \"ROS1\", \"SNAP25\", \"SSTR1\", \"TBX2\"}\n",
    "new_genes = [new_gene_names.get(g, g) for g in list(gene_exp_processed.columns)]\n",
    "gene_exp_processed.columns = new_genes\n",
    "missing_genes = set(gene_list[\"Symbol\"]) - set(gene_exp_processed.columns) - unmapped_genes\n",
    "missing_genes"
   ],
   "id": "10b888df4b3ddc6b",
   "outputs": [
    {
     "data": {
      "text/plain": [
       "set()"
      ]
     },
     "execution_count": 8,
     "metadata": {},
     "output_type": "execute_result"
    }
   ],
   "execution_count": 8
  },
  {
   "metadata": {
    "ExecuteTime": {
     "end_time": "2025-11-12T15:15:14.569208Z",
     "start_time": "2025-11-12T15:15:14.559895Z"
    }
   },
   "cell_type": "code",
   "source": [
    "gene_list2 = pd.read_csv('../gene_lists/drug_target_genes_all_drugs.csv')\n",
    "missing_genes2 = set(gene_list2[\"Symbol\"]) - set(gene_exp_processed.columns) - set(unmapped_genes)\n",
    "missing_genes2"
   ],
   "id": "5aa33663a5ee2738",
   "outputs": [
    {
     "data": {
      "text/plain": [
       "set()"
      ]
     },
     "execution_count": 9,
     "metadata": {},
     "output_type": "execute_result"
    }
   ],
   "execution_count": 9
  },
  {
   "metadata": {
    "ExecuteTime": {
     "end_time": "2025-11-12T15:35:10.372268Z",
     "start_time": "2025-11-12T15:35:10.260627Z"
    }
   },
   "cell_type": "code",
   "source": [
    "gene_list3 = pd.read_csv('../gene_lists/gene_list_paccmann_network_prop.csv')\n",
    "missing_genes3 = set(gene_list3[\"Symbol\"]) - set(gene_exp_processed.columns) - set(unmapped_genes)\n",
    "len(missing_genes3) # this is too many to map"
   ],
   "id": "d56727dbec33d310",
   "outputs": [
    {
     "data": {
      "text/plain": [
       "342"
      ]
     },
     "execution_count": 16,
     "metadata": {},
     "output_type": "execute_result"
    }
   ],
   "execution_count": 16
  },
  {
   "metadata": {
    "ExecuteTime": {
     "end_time": "2025-11-12T15:35:35.225138Z",
     "start_time": "2025-11-12T15:35:22.496747Z"
    }
   },
   "cell_type": "code",
   "source": "gene_exp_processed.to_csv('gene_expression/gene_expression.csv')",
   "id": "808576bfca0d3f1b",
   "outputs": [],
   "execution_count": 17
  },
  {
   "metadata": {},
   "cell_type": "markdown",
   "source": "## Mutation data",
   "id": "b23257dd404286ca"
  },
  {
   "metadata": {
    "ExecuteTime": {
     "end_time": "2025-11-12T16:25:31.106640Z",
     "start_time": "2025-11-12T16:25:30.990720Z"
    }
   },
   "cell_type": "code",
   "source": [
    "mut = pd.read_csv('mutation/beataml_wes_wv1to4_mutations_dbgap.txt', sep='\\t')\n",
    "mut.head()"
   ],
   "id": "eb23e4d09ed67726",
   "outputs": [
    {
     "data": {
      "text/plain": [
       "  dbgap_sample_id capture_type seqnames  pos_start    pos_end ref    alt  \\\n",
       "0         BA2336D  NexteraV1.2        4  106156042  106156043  TC      T   \n",
       "1         BA2336D  NexteraV1.2        4  106190829  106190830  AG      A   \n",
       "2         BA2336D  NexteraV1.2        5  170837543  170837543   C  CTCTG   \n",
       "3         BA2643D  NexteraV1.2       11   32456651   32456652  GC      G   \n",
       "4         BA2643D  NexteraV1.2        2   25457242   25457242   C      T   \n",
       "\n",
       "  genotyper  tumor_only  total_reads  ...   cdna_position  cds_position  \\\n",
       "0   varscan           1          151  ...      1804/10166      944/6009   \n",
       "1   varscan           1           74  ...      4968/10166     4108/6009   \n",
       "2   varscan           1           59  ...  1160-1161/1758   859-860/885   \n",
       "3   varscan           1           51  ...        525/3122      240/1554   \n",
       "4    mutect           1           28  ...       2983/4380     2645/2739   \n",
       "\n",
       "   protein_position  amino_acids       codons  \\\n",
       "0          315/2002          S/X       tCc/tc   \n",
       "1         1370/2002          G/X       Ggg/gg   \n",
       "2           287/294        L/LCX  ctc/cTCTGtc   \n",
       "3            80/517          L/X       ctG/ct   \n",
       "4           882/912          R/H      cGc/cAc   \n",
       "\n",
       "                 existing_variation                   refseq            sift  \\\n",
       "0                               NaN                      NaN             NaN   \n",
       "1                       rs756348991                      NaN             NaN   \n",
       "2            rs758959453&COSM158604              NM_002520.6             NaN   \n",
       "3                               NaN  NM_024426.4&NM_024424.3             NaN   \n",
       "4  rs147001633&COSM52944&COSM442676              NM_175629.2  deleterious(0)   \n",
       "\n",
       "                   polyphen   exac_af  \n",
       "0                       NaN       NaN  \n",
       "1                       NaN  0.000037  \n",
       "2                       NaN  0.000008  \n",
       "3                       NaN       NaN  \n",
       "4  probably_damaging(0.993)  0.000593  \n",
       "\n",
       "[5 rows x 32 columns]"
      ],
      "text/html": [
       "<div>\n",
       "<style scoped>\n",
       "    .dataframe tbody tr th:only-of-type {\n",
       "        vertical-align: middle;\n",
       "    }\n",
       "\n",
       "    .dataframe tbody tr th {\n",
       "        vertical-align: top;\n",
       "    }\n",
       "\n",
       "    .dataframe thead th {\n",
       "        text-align: right;\n",
       "    }\n",
       "</style>\n",
       "<table border=\"1\" class=\"dataframe\">\n",
       "  <thead>\n",
       "    <tr style=\"text-align: right;\">\n",
       "      <th></th>\n",
       "      <th>dbgap_sample_id</th>\n",
       "      <th>capture_type</th>\n",
       "      <th>seqnames</th>\n",
       "      <th>pos_start</th>\n",
       "      <th>pos_end</th>\n",
       "      <th>ref</th>\n",
       "      <th>alt</th>\n",
       "      <th>genotyper</th>\n",
       "      <th>tumor_only</th>\n",
       "      <th>total_reads</th>\n",
       "      <th>...</th>\n",
       "      <th>cdna_position</th>\n",
       "      <th>cds_position</th>\n",
       "      <th>protein_position</th>\n",
       "      <th>amino_acids</th>\n",
       "      <th>codons</th>\n",
       "      <th>existing_variation</th>\n",
       "      <th>refseq</th>\n",
       "      <th>sift</th>\n",
       "      <th>polyphen</th>\n",
       "      <th>exac_af</th>\n",
       "    </tr>\n",
       "  </thead>\n",
       "  <tbody>\n",
       "    <tr>\n",
       "      <th>0</th>\n",
       "      <td>BA2336D</td>\n",
       "      <td>NexteraV1.2</td>\n",
       "      <td>4</td>\n",
       "      <td>106156042</td>\n",
       "      <td>106156043</td>\n",
       "      <td>TC</td>\n",
       "      <td>T</td>\n",
       "      <td>varscan</td>\n",
       "      <td>1</td>\n",
       "      <td>151</td>\n",
       "      <td>...</td>\n",
       "      <td>1804/10166</td>\n",
       "      <td>944/6009</td>\n",
       "      <td>315/2002</td>\n",
       "      <td>S/X</td>\n",
       "      <td>tCc/tc</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>1</th>\n",
       "      <td>BA2336D</td>\n",
       "      <td>NexteraV1.2</td>\n",
       "      <td>4</td>\n",
       "      <td>106190829</td>\n",
       "      <td>106190830</td>\n",
       "      <td>AG</td>\n",
       "      <td>A</td>\n",
       "      <td>varscan</td>\n",
       "      <td>1</td>\n",
       "      <td>74</td>\n",
       "      <td>...</td>\n",
       "      <td>4968/10166</td>\n",
       "      <td>4108/6009</td>\n",
       "      <td>1370/2002</td>\n",
       "      <td>G/X</td>\n",
       "      <td>Ggg/gg</td>\n",
       "      <td>rs756348991</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>0.000037</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>2</th>\n",
       "      <td>BA2336D</td>\n",
       "      <td>NexteraV1.2</td>\n",
       "      <td>5</td>\n",
       "      <td>170837543</td>\n",
       "      <td>170837543</td>\n",
       "      <td>C</td>\n",
       "      <td>CTCTG</td>\n",
       "      <td>varscan</td>\n",
       "      <td>1</td>\n",
       "      <td>59</td>\n",
       "      <td>...</td>\n",
       "      <td>1160-1161/1758</td>\n",
       "      <td>859-860/885</td>\n",
       "      <td>287/294</td>\n",
       "      <td>L/LCX</td>\n",
       "      <td>ctc/cTCTGtc</td>\n",
       "      <td>rs758959453&amp;COSM158604</td>\n",
       "      <td>NM_002520.6</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>0.000008</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>3</th>\n",
       "      <td>BA2643D</td>\n",
       "      <td>NexteraV1.2</td>\n",
       "      <td>11</td>\n",
       "      <td>32456651</td>\n",
       "      <td>32456652</td>\n",
       "      <td>GC</td>\n",
       "      <td>G</td>\n",
       "      <td>varscan</td>\n",
       "      <td>1</td>\n",
       "      <td>51</td>\n",
       "      <td>...</td>\n",
       "      <td>525/3122</td>\n",
       "      <td>240/1554</td>\n",
       "      <td>80/517</td>\n",
       "      <td>L/X</td>\n",
       "      <td>ctG/ct</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NM_024426.4&amp;NM_024424.3</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>4</th>\n",
       "      <td>BA2643D</td>\n",
       "      <td>NexteraV1.2</td>\n",
       "      <td>2</td>\n",
       "      <td>25457242</td>\n",
       "      <td>25457242</td>\n",
       "      <td>C</td>\n",
       "      <td>T</td>\n",
       "      <td>mutect</td>\n",
       "      <td>1</td>\n",
       "      <td>28</td>\n",
       "      <td>...</td>\n",
       "      <td>2983/4380</td>\n",
       "      <td>2645/2739</td>\n",
       "      <td>882/912</td>\n",
       "      <td>R/H</td>\n",
       "      <td>cGc/cAc</td>\n",
       "      <td>rs147001633&amp;COSM52944&amp;COSM442676</td>\n",
       "      <td>NM_175629.2</td>\n",
       "      <td>deleterious(0)</td>\n",
       "      <td>probably_damaging(0.993)</td>\n",
       "      <td>0.000593</td>\n",
       "    </tr>\n",
       "  </tbody>\n",
       "</table>\n",
       "<p>5 rows Ã— 32 columns</p>\n",
       "</div>"
      ]
     },
     "execution_count": 35,
     "metadata": {},
     "output_type": "execute_result"
    }
   ],
   "execution_count": 35
  },
  {
   "metadata": {
    "ExecuteTime": {
     "end_time": "2025-11-12T16:25:40.140385Z",
     "start_time": "2025-11-12T16:25:40.110678Z"
    }
   },
   "cell_type": "code",
   "source": [
    "# we only want mutation yes/no per gene\n",
    "mut = mut[['dbgap_sample_id', 'symbol']].drop_duplicates()\n",
    "# merge with mapping\n",
    "mut = mut.merge(mapping[['dbgap_subject_id', 'dbgap_dnaseq_sample']], left_on='dbgap_sample_id', right_on='dbgap_dnaseq_sample', how='left')\n",
    "mut = mut[['dbgap_subject_id', 'symbol']].drop_duplicates()\n",
    "mut.columns = ['cell_line_name', 'symbol']\n",
    "mut['cell_line_name'] = mut['cell_line_name'].astype(str)\n",
    "unique_mut_samples = set(mut['cell_line_name'])\n",
    "print(f\"Mutation data missing for {len(unique_response_samples - unique_mut_samples)}/{(len(unique_response_samples))} response samples\")\n",
    "print(f\"Response data missing for {len(unique_mut_samples - unique_response_samples)}/{(len(unique_mut_samples))} mutation samples\")"
   ],
   "id": "2946d6c7a60c5f50",
   "outputs": [
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "Mutation data missing for 39/569 response samples\n",
      "Response data missing for 226/756 mutation samples\n"
     ]
    }
   ],
   "execution_count": 36
  },
  {
   "metadata": {
    "ExecuteTime": {
     "end_time": "2025-11-12T16:25:44.535110Z",
     "start_time": "2025-11-12T16:25:44.512845Z"
    }
   },
   "cell_type": "code",
   "source": [
    "# filter to retain only cell_line_names in intersection of unique_response_samples, unique_mut_samples\n",
    "mut = mut.set_index('cell_line_name')\n",
    "mut = mut.loc[list(unique_response_samples & unique_mut_samples)]"
   ],
   "id": "242ea6ddd07c9eb3",
   "outputs": [],
   "execution_count": 37
  },
  {
   "metadata": {
    "ExecuteTime": {
     "end_time": "2025-11-12T16:26:20.455957Z",
     "start_time": "2025-11-12T16:26:20.172337Z"
    }
   },
   "cell_type": "code",
   "source": [
    "wide_mut_df = (\n",
    "    mut.reset_index().assign(value=True)\n",
    "      .pivot(index='cell_line_name', columns='symbol', values='value')\n",
    "      .fillna(False)\n",
    "      .astype(bool)\n",
    ")\n",
    "wide_mut_df.iloc[0:5, 0:5]"
   ],
   "id": "6db4c73545fe8802",
   "outputs": [
    {
     "name": "stderr",
     "output_type": "stream",
     "text": [
      "/var/folders/3x/f8j9tddj7flfxt9zx1gkws1m0000gn/T/ipykernel_62253/3154144439.py:4: FutureWarning: Downcasting object dtype arrays on .fillna, .ffill, .bfill is deprecated and will change in a future version. Call result.infer_objects(copy=False) instead. To opt-in to the future behavior, set `pd.set_option('future.no_silent_downcasting', True)`\n",
      "  .fillna(False)\n"
     ]
    },
    {
     "data": {
      "text/plain": [
       "symbol           A1CF    A2M  A4GALT  AADAC  AADACL3\n",
       "cell_line_name                                      \n",
       "2003            False  False   False  False    False\n",
       "2005            False  False   False  False    False\n",
       "2007            False  False   False  False    False\n",
       "2008            False  False   False  False    False\n",
       "2010            False  False   False  False    False"
      ],
      "text/html": [
       "<div>\n",
       "<style scoped>\n",
       "    .dataframe tbody tr th:only-of-type {\n",
       "        vertical-align: middle;\n",
       "    }\n",
       "\n",
       "    .dataframe tbody tr th {\n",
       "        vertical-align: top;\n",
       "    }\n",
       "\n",
       "    .dataframe thead th {\n",
       "        text-align: right;\n",
       "    }\n",
       "</style>\n",
       "<table border=\"1\" class=\"dataframe\">\n",
       "  <thead>\n",
       "    <tr style=\"text-align: right;\">\n",
       "      <th>symbol</th>\n",
       "      <th>A1CF</th>\n",
       "      <th>A2M</th>\n",
       "      <th>A4GALT</th>\n",
       "      <th>AADAC</th>\n",
       "      <th>AADACL3</th>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>cell_line_name</th>\n",
       "      <th></th>\n",
       "      <th></th>\n",
       "      <th></th>\n",
       "      <th></th>\n",
       "      <th></th>\n",
       "    </tr>\n",
       "  </thead>\n",
       "  <tbody>\n",
       "    <tr>\n",
       "      <th>2003</th>\n",
       "      <td>False</td>\n",
       "      <td>False</td>\n",
       "      <td>False</td>\n",
       "      <td>False</td>\n",
       "      <td>False</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>2005</th>\n",
       "      <td>False</td>\n",
       "      <td>False</td>\n",
       "      <td>False</td>\n",
       "      <td>False</td>\n",
       "      <td>False</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>2007</th>\n",
       "      <td>False</td>\n",
       "      <td>False</td>\n",
       "      <td>False</td>\n",
       "      <td>False</td>\n",
       "      <td>False</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>2008</th>\n",
       "      <td>False</td>\n",
       "      <td>False</td>\n",
       "      <td>False</td>\n",
       "      <td>False</td>\n",
       "      <td>False</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>2010</th>\n",
       "      <td>False</td>\n",
       "      <td>False</td>\n",
       "      <td>False</td>\n",
       "      <td>False</td>\n",
       "      <td>False</td>\n",
       "    </tr>\n",
       "  </tbody>\n",
       "</table>\n",
       "</div>"
      ]
     },
     "execution_count": 39,
     "metadata": {},
     "output_type": "execute_result"
    }
   ],
   "execution_count": 39
  },
  {
   "metadata": {
    "ExecuteTime": {
     "end_time": "2025-11-12T16:33:58.222992Z",
     "start_time": "2025-11-12T16:33:58.211921Z"
    }
   },
   "cell_type": "code",
   "source": [
    "print(f'Missing landmark genes: {len(set(gene_list[\"Symbol\"]) - set(wide_mut_df.columns))}/{len(gene_list)}')\n",
    "print(f'Missing drug target genes: {len(set(gene_list2[\"Symbol\"]) - set(wide_mut_df.columns))}/{len(gene_list2)}')\n",
    "print(f'Missing paccman genes: {len(set(gene_list3[\"Symbol\"]) - set(wide_mut_df.columns))}/{len(gene_list3)}')"
   ],
   "id": "edb7a26ccd078427",
   "outputs": [
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "Missing landmark genes: 760/867\n",
      "Missing drug target genes: 196/242\n",
      "Missing paccman genes: 1701/1957\n"
     ]
    }
   ],
   "execution_count": 47
  },
  {
   "metadata": {
    "ExecuteTime": {
     "end_time": "2025-11-12T16:29:30.986933Z",
     "start_time": "2025-11-12T16:29:30.438065Z"
    }
   },
   "cell_type": "code",
   "source": "wide_mut_df.to_csv('mutation/mutations.csv')",
   "id": "4dd594ed64803a29",
   "outputs": [],
   "execution_count": 42
  },
  {
   "metadata": {},
   "cell_type": "markdown",
   "source": "## Other files",
   "id": "1a0250856cdf0334"
  },
  {
   "metadata": {
    "ExecuteTime": {
     "end_time": "2025-11-12T17:52:21.077413Z",
     "start_time": "2025-11-12T17:52:21.042738Z"
    }
   },
   "cell_type": "code",
   "source": [
    "# create cell line names file\n",
    "cell_line_names = response.drop_duplicates(\"cell_line_name\")[[\"cell_line_name\"]]\n",
    "cell_line_names[\"tissue\"] = \"Blood\"\n",
    "cell_line_names.to_csv(\"cell_line_names.csv\", index=False)"
   ],
   "id": "94df591d679b65d6",
   "outputs": [],
   "execution_count": 63
  },
  {
   "metadata": {
    "ExecuteTime": {
     "end_time": "2025-11-12T17:52:21.514772Z",
     "start_time": "2025-11-12T17:52:21.506876Z"
    }
   },
   "cell_type": "code",
   "source": [
    "# create drug names file\n",
    "drug_names = response.drop_duplicates(\"pubchem_id\")[[\"pubchem_id\", \"drug_name\"]]\n",
    "drug_names.to_csv(\"drug_names.csv\", index=False)"
   ],
   "id": "9641d5b98fa7d88e",
   "outputs": [],
   "execution_count": 64
  },
  {
   "metadata": {},
   "cell_type": "code",
   "outputs": [],
   "execution_count": null,
   "source": "",
   "id": "2c2bb735783724c8"
  }
 ],
 "metadata": {
  "kernelspec": {
   "display_name": "Python 3",
   "language": "python",
   "name": "python3"
  },
  "language_info": {
   "codemirror_mode": {
    "name": "ipython",
    "version": 2
   },
   "file_extension": ".py",
   "mimetype": "text/x-python",
   "name": "python",
   "nbconvert_exporter": "python",
   "pygments_lexer": "ipython2",
   "version": "2.7.6"
  }
 },
 "nbformat": 4,
 "nbformat_minor": 5
}
