{
 "cells": [
  {
   "cell_type": "code",
   "id": "initial_id",
   "metadata": {
    "collapsed": true,
    "ExecuteTime": {
     "end_time": "2025-11-13T09:29:43.859026Z",
     "start_time": "2025-11-13T09:29:43.448267Z"
    }
   },
   "source": "import pandas as pd",
   "outputs": [],
   "execution_count": 1
  },
  {
   "metadata": {},
   "cell_type": "markdown",
   "source": [
    "## Expression data\n",
    "\n",
    "\"PDTCs derived from ten of the PDTX models were extensively characterized using WES, sWGS, and RNAexp (microarrays).\"\n",
    "\n",
    "\"The PDTX samples were comprehensively molecularly characterized at several passages using sWGS (for CNAs), WES (for single nucleotide variations [SNVs]), reduced-representation bisulfite sequencing (“RRBS”) (for DNA methylation), and RNAexp (for global expression and pathway activity profiling).\"\n",
    "\n",
    "-> there are more microarray samples because they also cover PDTX, not only PDTCs."
   ],
   "id": "6e79a3b07a41a821"
  },
  {
   "metadata": {
    "ExecuteTime": {
     "end_time": "2025-11-13T09:29:46.705234Z",
     "start_time": "2025-11-13T09:29:46.348413Z"
    }
   },
   "cell_type": "code",
   "source": [
    "microarray = pd.read_csv('gene_expression/ExpressionSamples.txt', sep='\\t', index_col=0)\n",
    "microarray.shape"
   ],
   "id": "bc6b164cd66fce90",
   "outputs": [
    {
     "data": {
      "text/plain": [
       "(22044, 153)"
      ]
     },
     "execution_count": 2,
     "metadata": {},
     "output_type": "execute_result"
    }
   ],
   "execution_count": 2
  },
  {
   "metadata": {
    "ExecuteTime": {
     "end_time": "2025-11-13T09:29:50.255508Z",
     "start_time": "2025-11-13T09:29:50.221917Z"
    }
   },
   "cell_type": "code",
   "source": [
    "# check response overlap\n",
    "responses = pd.read_csv('response/Bruna.csv')\n",
    "print(f'All responses samples {len(set(responses[\"cell_line_name\"]))}')\n",
    "print(f'All expression samples {len(set(microarray.columns))}')\n",
    "print(f'Intersection {len(set(responses[\"cell_line_name\"]).intersection(set(microarray.columns)))}')\n",
    "print(set(responses[\"cell_line_name\"]).intersection(set(microarray.columns)))"
   ],
   "id": "3924956aae51d0f",
   "outputs": [
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "All responses samples 37\n",
      "All expression samples 153\n",
      "Intersection 3\n",
      "{'HCI009-X4C', 'STG282-X1C', 'STG282-X1CR'}\n"
     ]
    }
   ],
   "execution_count": 3
  },
  {
   "metadata": {
    "ExecuteTime": {
     "end_time": "2025-11-13T09:29:58.475888Z",
     "start_time": "2025-11-13T09:29:58.437093Z"
    }
   },
   "cell_type": "code",
   "source": [
    "### Export cell line names and drug names\n",
    "cell_line_names = set(responses['cell_line_name'])\n",
    "drug_names = responses[['pubchem_id', 'drug_name']].drop_duplicates()\n",
    "cell_line_name_df = pd.DataFrame({'cell_line_name': list(cell_line_names), 'tissue': 'Breast'})\n",
    "\n",
    "drug_names.to_csv('drug_names.csv', index=False)\n",
    "cell_line_name_df.to_csv('cell_line_names.csv', index=False)"
   ],
   "id": "ca54cd95e500485c",
   "outputs": [],
   "execution_count": 4
  },
  {
   "metadata": {
    "ExecuteTime": {
     "end_time": "2025-11-13T09:29:59.154192Z",
     "start_time": "2025-11-13T09:29:59.150683Z"
    }
   },
   "cell_type": "code",
   "source": [
    "all_resp_samples = pd.Series(list(responses[\"cell_line_name\"].unique()))\n",
    "all_micro_samples = pd.Series(list(microarray.columns))"
   ],
   "id": "19ebaad8f5b51f66",
   "outputs": [],
   "execution_count": 5
  },
  {
   "metadata": {
    "ExecuteTime": {
     "end_time": "2025-11-13T09:30:02.038274Z",
     "start_time": "2025-11-13T09:30:02.013756Z"
    }
   },
   "cell_type": "code",
   "source": [
    "import re\n",
    "# function to extract model and passage only\n",
    "def normalize_id(x):\n",
    "    m = re.match(r'([A-Z0-9]+-X\\d+)', x)\n",
    "    return m.group(1) if m else x\n",
    "\n",
    "resp_key = all_resp_samples.apply(normalize_id)\n",
    "micro_key = all_micro_samples.apply(normalize_id)\n",
    "\n",
    "# find matches\n",
    "matches = pd.merge(\n",
    "    pd.DataFrame({'resp': all_resp_samples, 'key': resp_key}),\n",
    "    pd.DataFrame({'micro': all_micro_samples, 'key': micro_key}),\n",
    "    on='key',\n",
    "    how='left'\n",
    ")"
   ],
   "id": "f040bd80bcd268cd",
   "outputs": [],
   "execution_count": 6
  },
  {
   "metadata": {
    "ExecuteTime": {
     "end_time": "2025-11-13T09:30:04.951740Z",
     "start_time": "2025-11-13T09:30:04.919889Z"
    }
   },
   "cell_type": "code",
   "source": [
    "mapping = {\n",
    "    \"HCI001-X4\": \"HCI001-X4C\",\n",
    "    \"HCI001-X5\": \"HCI001-X5C\",\n",
    "    \"HCI002-X2\": \"HCI002-X2C\",\n",
    "    # no microarray match for \"HCI002-X2CR\"\n",
    "    \"HCI002-X3R\": \"HCI002-X3C\", # might be a bad match?\n",
    "    \"HCI002-X6\": \"HCI002-X6C\",\n",
    "    \"HCI005-X2\": \"HCI005-X2C\",\n",
    "    \"HCI008-X1\": \"HCI008-X1C\",\n",
    "    \"HCI009-X1\": \"HCI009-X1C\",\n",
    "    \"HCI009-X4C\": \"HCI009-X4C\", # exact match\n",
    "    # no microarray match for \"HCI009-X7C\"\n",
    "    \"HCI010-X2\": \"HCI010-X2C\",\n",
    "    \"HCI010-X2R\": \"HCI010-X2CR\",\n",
    "    \"HCI011-X1\": \"HCI011-X1C\",\n",
    "    \"IC007-X0\": \"IC007-X0C\",\n",
    "    \"STG139-X13\": \"STG139-X13C\",\n",
    "    # no matches for \"STG139-X9C\", \"STG139M-X4C\n",
    "    \"STG139M-X5\": \"STG139M-X5C\",\n",
    "    \"STG143-X2\": \"STG143-X2C\",\n",
    "    \"STG195-X3\": \"STG195-X3C\",\n",
    "    \"STG201-X3\": \"STG201-X3C\",\n",
    "    \"STG282-X1C\": \"STG282-X1C\", # exact match\n",
    "    \"STG282-X1CR\": \"STG282-X1CR\", # exact match\n",
    "    \"STG282-X1R1\": \"STG282-X1CR1\",\n",
    "    \"STG282-X3\": \"STG282-X3C\",\n",
    "    \"STG316-X1\": \"STG316-X1C\",\n",
    "    \"STG316-X1R\": \"STG316-X1CR\",\n",
    "    \"STG335-X1\": \"STG335-X1C\",\n",
    "    # no matches for \"STG335-X2C\", \"VHIO098-X1C\", \"VHIO169-X11C\", \"VHIO169-X3C\",\n",
    "    # \"VHIO179-X1C\", \"VHIO244-X1C\", \"VHIO244-X1CR\", \"VHIO244-X2C\"\n",
    "}\n",
    "# missing: 12/37 -> 25 matches\n",
    "# rename microarray sample names according to mapping\n",
    "microarray = microarray.rename(columns=mapping)\n",
    "print(f'Intersection {len(set(responses[\"cell_line_name\"]).intersection(set(microarray.columns)))}')\n",
    "# only keep intersecting columns\n",
    "microarray = microarray.loc[:, list(set(responses[\"cell_line_name\"]).intersection(set(microarray.columns)))]"
   ],
   "id": "98444fbcd71c4c53",
   "outputs": [
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "Intersection 25\n"
     ]
    }
   ],
   "execution_count": 7
  },
  {
   "metadata": {
    "ExecuteTime": {
     "end_time": "2025-11-13T09:30:05.846874Z",
     "start_time": "2025-11-13T09:30:05.821783Z"
    }
   },
   "cell_type": "code",
   "source": [
    "microarray_transposed = microarray.T\n",
    "microarray_transposed = microarray_transposed.reset_index().rename(columns={'index': 'cell_line_name'})\n",
    "microarray_transposed = microarray_transposed.set_index(['cell_line_name'])"
   ],
   "id": "14de7d0b19f5b3ae",
   "outputs": [],
   "execution_count": 8
  },
  {
   "metadata": {
    "ExecuteTime": {
     "end_time": "2025-11-13T09:30:08.129791Z",
     "start_time": "2025-11-13T09:30:08.110741Z"
    }
   },
   "cell_type": "code",
   "source": [
    "gene_names = pd.DataFrame({'Symbol': microarray_transposed.columns})\n",
    "gene_names.to_csv('gene_expression/gene_names.csv', index=False)"
   ],
   "id": "3c93f80e7216c7f6",
   "outputs": [],
   "execution_count": 9
  },
  {
   "metadata": {
    "ExecuteTime": {
     "end_time": "2025-11-13T09:30:42.285079Z",
     "start_time": "2025-11-13T09:30:42.234191Z"
    }
   },
   "cell_type": "code",
   "source": [
    "gene_list = pd.read_csv('../gene_lists/landmark_genes.csv', index_col=0)\n",
    "\n",
    "# map genes\n",
    "new_gene_names = {\n",
    "    \"AARS\": \"AARS1\",\n",
    "    \"SPATA5L1\": \"AFG2B\",\n",
    "    \"FAM123A\": \"AMER2\",\n",
    "    \"C20orf3\": \"APMAP\",\n",
    "    \"KIAA0100\": \"BLTP2\",\n",
    "    \"UHRF1BP1L\": \"BLTP3B\",\n",
    "    \"C20orf114\": \"BPIFB1\",\n",
    "    \"KIAA2026\": \"BRD10\",\n",
    "    \"BACH1\": \"BRIP1\",\n",
    "    \"KIAA0528\": \"C2CD5\",\n",
    "    \"CAMSAP1L1\": \"CAMSAP2\",\n",
    "    \"TMEM146\": \"CATSPERD\",\n",
    "    \"C6orf97\": \"CCDC170\",\n",
    "    \"WISP1\": \"CCN4\",\n",
    "    \"CP110\": \"CCP110\",\n",
    "    \"FAM190A\": \"CCSER1\",\n",
    "    \"C16orf5\": \"CDIP1\",\n",
    "    \"NARFL\": \"CIAO3\",\n",
    "    \"ODF3L2\": \"CIMAP1D\",\n",
    "    \"CLECL1\": \"CLECL1P\",\n",
    "    \"PPPDE2\": \"DESI1\",\n",
    "    \"FAM69A\": \"DIPK1A\",\n",
    "    \"FAM69C\": \"DIPK1C\",\n",
    "    \"TCTEX1D4\": \"DYNLT4\",\n",
    "    \"FAM102B\": \"EEIG2\",\n",
    "    \"C3orf25\": \"EFCAB12\",\n",
    "    \"KIAA0494\": \"EFCAB14\",\n",
    "    \"CCDC48\": \"EFCC1\",\n",
    "    \"C15orf24\": \"EMC7\",\n",
    "    \"C19orf63\": \"EMC10\",\n",
    "    \"EPRS\": \"EPRS1\",\n",
    "    \"C10orf84\": \"FAM204A\",\n",
    "    \"C20orf108\": \"FAM210B\",\n",
    "    \"C9orf25\": \"FAM219A\",\n",
    "    \"ALS2CR12\": \"FLACC1\",\n",
    "    \"KIAA0355\": \"GARRE1\",\n",
    "    \"WRB\": \"GET1\",\n",
    "    \"TSTA3\": \"GFUS\",\n",
    "    \"H2AFV\": \"H2AZ2\",\n",
    "    \"HIST1H2BK\": \"H2BC12\",\n",
    "    \"GPR81\": \"HCAR1\",\n",
    "    \"IARS\": \"IARS1\",\n",
    "    \"IL1F7\": \"IL37\",\n",
    "    \"SSFA2\": \"ITPRID2\",\n",
    "    \"MYST3\": \"KAT6A\",\n",
    "    \"MYST4\": \"KAT6B\",\n",
    "    \"KIAA1267\": \"KANSL1\",\n",
    "    \"C2orf67\": \"KANSL1L\",\n",
    "    \"KIAA0556\": \"KATNIP\",\n",
    "    \"KIAA1004\": \"KDM2A\",\n",
    "    \"C6orf221\": \"KHDC3L\",\n",
    "    \"HK2\": \"KLK2\",\n",
    "    \"GRP\": \"LGALSL\",\n",
    "    \"LOR\": \"LORICRIN\",\n",
    "    \"MARCH3\": \"MARCHF3\",\n",
    "    \"C17orf37\": \"MIEN1\",\n",
    "    \"RAGE\": \"MOK\",\n",
    "    \"C20orf20\": \"MRGBP\",\n",
    "    \"BRP44\": \"MPC2\",\n",
    "    \"HEATR7A\": \"MROH1\",\n",
    "    \"C1orf129\": \"MROH9\",\n",
    "    \"MAL\": \"MRTFA\",\n",
    "    \"MRPL15\": \"MRPL19\",\n",
    "    \"MOSC2\": \"MTARC2\",\n",
    "    \"C20orf7\": \"NDUFAF5\",\n",
    "    \"METTL11A\": \"NTMT1\",\n",
    "    \"KIAA1486\": \"NYAP2\",\n",
    "    \"ARMC4\": \"ODAD2\",\n",
    "    \"C12orf48\": \"PARPBP\",\n",
    "    \"GPR44\": \"PTGDR2\",\n",
    "    \"QARS\": \"QARS1\",\n",
    "    \"RAD51L3\": \"RAD51D\",\n",
    "    \"C9orf23\": \"RPP25L\",\n",
    "    \"C1orf113\": \"SH3D21\",\n",
    "    \"SKIV2L\": \"SKIC2\",\n",
    "    \"WDR61\": \"SKIC8\",\n",
    "    \"DIRC2\": \"SLC49A4\",\n",
    "    \"ANKRD43\": \"SOWAHA\",\n",
    "    \"FAM75D1\": \"SPATA31D1\",\n",
    "    \"C17orf46\": \"SPATA32\",\n",
    "    \"C1orf124\": \"SPRTN\",\n",
    "    \"FAM48A\": \"SUPT20H\",\n",
    "    \"C19orf39\": \"SWSAP1\",\n",
    "    \"FAM19A2\": \"TAFA2\",\n",
    "    \"TARSL2\": \"TARS3\",\n",
    "    \"TCP10\": \"TCP10L3\",\n",
    "    \"ODZ2\": \"TENM2\",\n",
    "    \"PAPD7\": \"TENT4A\",\n",
    "    \"C13orf27\": \"TEX30\",\n",
    "    \"FAM57A\": \"TLCD3A\",\n",
    "    \"FAM57B\": \"TLCD3B\",\n",
    "    \"FAM70B\": \"TMEM255B\",\n",
    "    \"C19orf6\": \"TMEM259\",\n",
    "    \"C3orf77\": \"TOPAZ1\",\n",
    "    \"C9orf167\": \"TOR4A\",\n",
    "    \"C18orf10\": \"TPGS2\",\n",
    "    \"C8orf83\": \"TRIQK\",\n",
    "    \"C3orf78\": \"UQCC5\",\n",
    "    \"PKM2\": \"PKM\",\n",
    "    \"VNN3\": \"VNN3P\",\n",
    "    \"WARS\": \"WARS1\",\n",
    "    \"ZFP161\": \"ZBTB14\",\n",
    "    \"FAM164C\": \"ZC2HC1C\",\n",
    "    \"PRF1\": \"ZNF395\",\n",
    "}\n",
    "unmapped_genes = {\"ADAT1\", \"APBB2\", \"ARFIP2\", \"CCDC85B\", \"CCNF\", \"CFLAR\", \"DNAJC15\", \"H2BC21\", \"HERC6\", \"HMGCR\", \"IKZF1\", \"MALT1\", \"MAPK13\", \"MKNK1\", \"MRPS16\", \"MYCBP\", \"NENF\", \"NFATC4\", \"NFKBIE\", \"NUP133\", \"PARP1\", \"POLR2K\", \"POP4\", \"PRSS23\", \"PXN\", \"RSU1\", \"RNMT\", \"SCARB1\", \"SNX6\", \"TLR4\", \"TSEN2\", \"UBE2J1\"}\n",
    "\n",
    "new_genes = [new_gene_names.get(g, g) for g in list(microarray_transposed.columns)]\n",
    "microarray_transposed.columns = new_genes\n",
    "missing_genes = set(gene_list[\"Symbol\"]) - set(microarray_transposed.columns) - set(unmapped_genes)\n",
    "missing_genes"
   ],
   "id": "c50f327fdc3b39da",
   "outputs": [
    {
     "data": {
      "text/plain": [
       "set()"
      ]
     },
     "execution_count": 11,
     "metadata": {},
     "output_type": "execute_result"
    }
   ],
   "execution_count": 11
  },
  {
   "metadata": {
    "ExecuteTime": {
     "end_time": "2025-11-13T09:30:48.890534Z",
     "start_time": "2025-11-13T09:30:48.868491Z"
    }
   },
   "cell_type": "code",
   "source": [
    "gene_list2 = pd.read_csv('../gene_lists/drug_target_genes_all_drugs.csv')\n",
    "unmapped_genes.update({\"CHUK\", \"EEF2K\", \"IRAK4\", \"MDM4\", \"MKNK1\", \"PARP1\", \"PIK3CG\"})\n",
    "missing_genes2 = set(gene_list2[\"Symbol\"]) - set(microarray_transposed.columns) - set(unmapped_genes)\n",
    "missing_genes2"
   ],
   "id": "71dac2d0a555c2ae",
   "outputs": [
    {
     "data": {
      "text/plain": [
       "set()"
      ]
     },
     "execution_count": 12,
     "metadata": {},
     "output_type": "execute_result"
    }
   ],
   "execution_count": 12
  },
  {
   "metadata": {
    "ExecuteTime": {
     "end_time": "2025-11-13T09:30:49.539741Z",
     "start_time": "2025-11-13T09:30:49.530301Z"
    }
   },
   "cell_type": "code",
   "source": [
    "gene_list3 = pd.read_csv('../gene_lists/gene_list_paccmann_network_prop.csv')\n",
    "unmapped_genes.update({\"AARD\", \"AFAP1L1\", \"AP5M1\", \"ATF7\", \"ATG12\", \"ATP10B\", \"C1RL\", \"CARD6\", \"CASR\", \"CDKN2AIPNL\", \"CEP97\", \"CHRNB1\", \"CLN5\", \"CYP1A2\", \"CYP27C1\", \"DCTN5\", \"DRAXIN\", \"DYDC2\", \"ENDOV\", \"EIF2S1\", \"FAM111B\", \"FAM216B\", \"FANCF\", \"FCRL4\", \"FGFBP3\", \"FHIT\", \"FMO2\", \"GALNT13\", \"GCFC2\", \"GPR160\", \"HAPLN4\", \"HAS1\", \"HHLA2\", \"HIF1AN\", \"HLA-DRB5\", \"IDI1\", \"IFT57\", \"IGFL3\", \"ILDR1\", \"KLHL23\", \"LONP2\", \"LSG1\", \"LYPD3\", \"MAST3\", \"MAST4\", \"MDFIC\", \"MED29\", \"MEGF8\", \"MFSD11\", \"NAALAD2\", \"NBL1\", \"NPW\", \"NRL\", \"OSGIN2\", \"PAPSS2\", \"PCDHB12\", \"PCYOX1L\", \"PEAR1\", \"POLR1G\", \"PRKRIP1\", \"PRR4\", \"RAB42\", \"RAB6B\", \"RHBDL2\", \"RUNDC3A\", \"S100A1\", \"SAMD9\", \"SH3BGRL2\", \"SIGLEC6\", \"SLC29A2\", \"SLFN13\", \"SPATA12\", \"SPRYD7\", \"STPG1\", \"SUV39H2\", \"TBX19\", \"TFCP2L1\", \"TLL2\", \"TMEM235\", \"TMEM86A\", \"TTC19\", \"TVP23A\", \"ULK2\", \"UNC45B\", \"UQCR11\", \"USP45\", \"WBP2NL\", \"ZFAND1\", \"ZFP1\", \"ZNF468\", \"ZNF490\", \"ZNF510\", \"ZNF600\", \"ZNF620\", \"ZNF71\", \"ZNF772\", \"ZSCAN22\"})\n",
    "missing_genes3 = set(gene_list3[\"Symbol\"]) - set(microarray_transposed.columns) - set(unmapped_genes)\n",
    "missing_genes3"
   ],
   "id": "16bdb152293754a2",
   "outputs": [
    {
     "data": {
      "text/plain": [
       "set()"
      ]
     },
     "execution_count": 13,
     "metadata": {},
     "output_type": "execute_result"
    }
   ],
   "execution_count": 13
  },
  {
   "metadata": {
    "ExecuteTime": {
     "end_time": "2025-11-13T09:30:51.871038Z",
     "start_time": "2025-11-13T09:30:51.343304Z"
    }
   },
   "cell_type": "code",
   "source": "microarray_transposed.to_csv('gene_expression/gene_expression.csv')",
   "id": "52f6524774693819",
   "outputs": [],
   "execution_count": 14
  },
  {
   "metadata": {},
   "cell_type": "markdown",
   "source": [
    "### Copy number variations\n",
    "\n",
    "sWGS"
   ],
   "id": "901f2760db5ecb05"
  },
  {
   "metadata": {
    "ExecuteTime": {
     "end_time": "2025-11-13T09:30:53.996352Z",
     "start_time": "2025-11-13T09:30:53.975515Z"
    }
   },
   "cell_type": "code",
   "source": "cnv = pd.read_csv('cnv/CNASamples.txt', sep='\\t')",
   "id": "ac13aee7dac7b56e",
   "outputs": [],
   "execution_count": 15
  },
  {
   "metadata": {
    "ExecuteTime": {
     "end_time": "2025-11-13T09:30:54.764699Z",
     "start_time": "2025-11-13T09:30:54.722694Z"
    }
   },
   "cell_type": "code",
   "source": [
    "cnv_key = pd.Series(cnv[\"ID\"].unique()).apply(normalize_id)\n",
    "\n",
    "# find matches\n",
    "matches = pd.merge(\n",
    "    pd.DataFrame({'resp': all_resp_samples, 'key': resp_key}),\n",
    "    pd.DataFrame({'cnv': cnv[\"ID\"].unique(), 'key': cnv_key}),\n",
    "    on='key',\n",
    "    how='left'\n",
    ")\n",
    "\n",
    "mapping = {\n",
    "    \"HCI001-X4\": \"HCI001-X4C\",\n",
    "    \"HCI001-X5\": \"HCI001-X5C\",\n",
    "    # no microarray match for \"HCI002-X2\", \"HCI002-X2CR\"\n",
    "    \"HCI002-X3R\": \"HCI002-X3C\", # might be a bad match?\n",
    "    \"HCI002-X6\": \"HCI002-X6C\",\n",
    "    \"HCI005-X2R\": \"HCI005-X2C\",  # might be a bad match?\n",
    "    \"HCI008-X1\": \"HCI008-X1C\",\n",
    "    \"HCI009-X1R\": \"HCI009-X1C\", # might be a bad match?\n",
    "    \"HCI009-X4C4\": \"HCI009-X4C\", # might be a bad match?\n",
    "    \"HCI009-X7\": \"HCI009-X7C\",\n",
    "    \"HCI010-X2\": \"HCI010-X2C\",\n",
    "    \"HCI010-X2R\": \"HCI010-X2CR\",\n",
    "    \"HCI011-X1\": \"HCI011-X1C\",\n",
    "    # no matches for \"IC007-X0C\"\n",
    "    \"STG139-X13\": \"STG139-X13C\",\n",
    "    # no matches for \"STG139-X9C\", \"STG139M-X4C\n",
    "    \"STG139M-X5\": \"STG139M-X5C\",\n",
    "    \"STG143-X2\": \"STG143-X2C\",\n",
    "    \"STG195-X3\": \"STG195-X3C\",\n",
    "    \"STG201-X3\": \"STG201-X3C\",\n",
    "    \"STG282-X1\": \"STG282-X1C\",\n",
    "    \"STG282-X1R\": \"STG282-X1CR\",\n",
    "    \"STG282-X1R1\": \"STG282-X1CR1\",\n",
    "    # no match for \"STG282-X3C\"\n",
    "    \"STG316-X1\": \"STG316-X1C\",\n",
    "    \"STG316-X1R\": \"STG316-X1CR\",\n",
    "    \"STG335-X1\": \"STG335-X1C\",\n",
    "    # no matches for \"STG335-X2C\", \"VHIO098-X1C\", \"VHIO169-X11C\", \"VHIO169-X3C\",\n",
    "    # \"VHIO179-X1C\", \"VHIO244-X1C\", \"VHIO244-X1CR\", \"VHIO244-X2C\"\n",
    "}\n",
    "\n",
    "# missing: 14/37 -> 23 matches\n",
    "# rename cnv sample names in cnv[\"ID\"] according to mapping\n",
    "cnv[\"cell_line_name\"] = cnv[\"ID\"].apply(lambda x: mapping.get(x, pd.NA))\n",
    "\n",
    "print(f'Intersection {len(set(responses[\"cell_line_name\"]).intersection(set(cnv[\"cell_line_name\"])))}')\n",
    "# only keep intersecting columns\n",
    "cnv = cnv[~pd.isna(cnv[\"cell_line_name\"])]\n",
    "cnv.head()"
   ],
   "id": "40959db17261324d",
   "outputs": [
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "Intersection 23\n"
     ]
    },
    {
     "data": {
      "text/plain": [
       "             ID chrom  loc.start    loc.end  num.mark  seg.mean  call  \\\n",
       "5294  HCI001-X4     1     900001   40600000       357    -0.599  HETD   \n",
       "5295  HCI001-X4     1   40600001  119400000       743     0.001  NEUT   \n",
       "5296  HCI001-X4     1  119400001  145700000        15     0.444  GAIN   \n",
       "5297  HCI001-X4     1  146500001  149200000         9     0.714  GAIN   \n",
       "5298  HCI001-X4     1  149900001  151500000        15     0.849   AMP   \n",
       "\n",
       "     cell_line_name  \n",
       "5294     HCI001-X4C  \n",
       "5295     HCI001-X4C  \n",
       "5296     HCI001-X4C  \n",
       "5297     HCI001-X4C  \n",
       "5298     HCI001-X4C  "
      ],
      "text/html": [
       "<div>\n",
       "<style scoped>\n",
       "    .dataframe tbody tr th:only-of-type {\n",
       "        vertical-align: middle;\n",
       "    }\n",
       "\n",
       "    .dataframe tbody tr th {\n",
       "        vertical-align: top;\n",
       "    }\n",
       "\n",
       "    .dataframe thead th {\n",
       "        text-align: right;\n",
       "    }\n",
       "</style>\n",
       "<table border=\"1\" class=\"dataframe\">\n",
       "  <thead>\n",
       "    <tr style=\"text-align: right;\">\n",
       "      <th></th>\n",
       "      <th>ID</th>\n",
       "      <th>chrom</th>\n",
       "      <th>loc.start</th>\n",
       "      <th>loc.end</th>\n",
       "      <th>num.mark</th>\n",
       "      <th>seg.mean</th>\n",
       "      <th>call</th>\n",
       "      <th>cell_line_name</th>\n",
       "    </tr>\n",
       "  </thead>\n",
       "  <tbody>\n",
       "    <tr>\n",
       "      <th>5294</th>\n",
       "      <td>HCI001-X4</td>\n",
       "      <td>1</td>\n",
       "      <td>900001</td>\n",
       "      <td>40600000</td>\n",
       "      <td>357</td>\n",
       "      <td>-0.599</td>\n",
       "      <td>HETD</td>\n",
       "      <td>HCI001-X4C</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>5295</th>\n",
       "      <td>HCI001-X4</td>\n",
       "      <td>1</td>\n",
       "      <td>40600001</td>\n",
       "      <td>119400000</td>\n",
       "      <td>743</td>\n",
       "      <td>0.001</td>\n",
       "      <td>NEUT</td>\n",
       "      <td>HCI001-X4C</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>5296</th>\n",
       "      <td>HCI001-X4</td>\n",
       "      <td>1</td>\n",
       "      <td>119400001</td>\n",
       "      <td>145700000</td>\n",
       "      <td>15</td>\n",
       "      <td>0.444</td>\n",
       "      <td>GAIN</td>\n",
       "      <td>HCI001-X4C</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>5297</th>\n",
       "      <td>HCI001-X4</td>\n",
       "      <td>1</td>\n",
       "      <td>146500001</td>\n",
       "      <td>149200000</td>\n",
       "      <td>9</td>\n",
       "      <td>0.714</td>\n",
       "      <td>GAIN</td>\n",
       "      <td>HCI001-X4C</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>5298</th>\n",
       "      <td>HCI001-X4</td>\n",
       "      <td>1</td>\n",
       "      <td>149900001</td>\n",
       "      <td>151500000</td>\n",
       "      <td>15</td>\n",
       "      <td>0.849</td>\n",
       "      <td>AMP</td>\n",
       "      <td>HCI001-X4C</td>\n",
       "    </tr>\n",
       "  </tbody>\n",
       "</table>\n",
       "</div>"
      ]
     },
     "execution_count": 16,
     "metadata": {},
     "output_type": "execute_result"
    }
   ],
   "execution_count": 16
  },
  {
   "metadata": {
    "ExecuteTime": {
     "end_time": "2025-11-13T09:30:55.682921Z",
     "start_time": "2025-11-13T09:30:55.664237Z"
    }
   },
   "cell_type": "code",
   "source": [
    "# remove call (not needed for GISTIC reprocessing), save\n",
    "cnv = cnv.drop(columns=['call', 'ID'])\n",
    "cnv = cnv.set_index('cell_line_name')\n",
    "cnv.to_csv('cnv/bruna_cnv_for_gistic.seg.txt', sep='\\t')"
   ],
   "id": "279276ab3a40acac",
   "outputs": [],
   "execution_count": 17
  },
  {
   "metadata": {},
   "cell_type": "markdown",
   "source": "Upload to gistic, use Human Hg19 reference genome with default settings.",
   "id": "9d737018702ec9d6"
  },
  {
   "metadata": {
    "ExecuteTime": {
     "end_time": "2025-11-13T09:31:02.612654Z",
     "start_time": "2025-11-13T09:31:02.553626Z"
    }
   },
   "cell_type": "code",
   "source": [
    "cnv_processed = pd.read_csv('cnv/all_thresholded.by_genes.txt', sep='\\t')\n",
    "cnv_processed = cnv_processed.set_index(\"Gene Symbol\")\n",
    "cnv_processed = cnv_processed.drop(columns=['Locus ID', 'Cytoband'])\n",
    "cnv_processed = cnv_processed.T\n",
    "cnv_processed.index.name = 'cell_line_name'\n",
    "# sort columns alphabetically\n",
    "cnv_processed = cnv_processed.reindex(sorted(cnv_processed.columns), axis=1)\n",
    "cnv_processed = cnv_processed.reset_index()\n",
    "cnv_processed.iloc[0:5, 0:5]"
   ],
   "id": "d230454bfeae67e7",
   "outputs": [
    {
     "data": {
      "text/plain": [
       "Gene Symbol cell_line_name  1/2-SBSRNA4  A1BG  A1BG-AS1  A1CF\n",
       "0               HCI001-X4C           -1    -1        -1     1\n",
       "1               HCI001-X5C           -1    -1        -1     1\n",
       "2               HCI002-X3C            0    -1        -1     0\n",
       "3               HCI002-X6C            0     0         0     0\n",
       "4               HCI005-X2C           -1     1         1    -1"
      ],
      "text/html": [
       "<div>\n",
       "<style scoped>\n",
       "    .dataframe tbody tr th:only-of-type {\n",
       "        vertical-align: middle;\n",
       "    }\n",
       "\n",
       "    .dataframe tbody tr th {\n",
       "        vertical-align: top;\n",
       "    }\n",
       "\n",
       "    .dataframe thead th {\n",
       "        text-align: right;\n",
       "    }\n",
       "</style>\n",
       "<table border=\"1\" class=\"dataframe\">\n",
       "  <thead>\n",
       "    <tr style=\"text-align: right;\">\n",
       "      <th>Gene Symbol</th>\n",
       "      <th>cell_line_name</th>\n",
       "      <th>1/2-SBSRNA4</th>\n",
       "      <th>A1BG</th>\n",
       "      <th>A1BG-AS1</th>\n",
       "      <th>A1CF</th>\n",
       "    </tr>\n",
       "  </thead>\n",
       "  <tbody>\n",
       "    <tr>\n",
       "      <th>0</th>\n",
       "      <td>HCI001-X4C</td>\n",
       "      <td>-1</td>\n",
       "      <td>-1</td>\n",
       "      <td>-1</td>\n",
       "      <td>1</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>1</th>\n",
       "      <td>HCI001-X5C</td>\n",
       "      <td>-1</td>\n",
       "      <td>-1</td>\n",
       "      <td>-1</td>\n",
       "      <td>1</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>2</th>\n",
       "      <td>HCI002-X3C</td>\n",
       "      <td>0</td>\n",
       "      <td>-1</td>\n",
       "      <td>-1</td>\n",
       "      <td>0</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>3</th>\n",
       "      <td>HCI002-X6C</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>4</th>\n",
       "      <td>HCI005-X2C</td>\n",
       "      <td>-1</td>\n",
       "      <td>1</td>\n",
       "      <td>1</td>\n",
       "      <td>-1</td>\n",
       "    </tr>\n",
       "  </tbody>\n",
       "</table>\n",
       "</div>"
      ]
     },
     "execution_count": 18,
     "metadata": {},
     "output_type": "execute_result"
    }
   ],
   "execution_count": 18
  },
  {
   "metadata": {
    "ExecuteTime": {
     "end_time": "2025-11-13T09:31:05.720503Z",
     "start_time": "2025-11-13T09:31:05.711989Z"
    }
   },
   "cell_type": "code",
   "source": [
    "# sanity check\n",
    "new_genes = [new_gene_names.get(g, g) for g in list(cnv_processed.columns)]\n",
    "cnv_processed.columns = new_genes"
   ],
   "id": "636950a095e0bbba",
   "outputs": [],
   "execution_count": 19
  },
  {
   "metadata": {
    "ExecuteTime": {
     "end_time": "2025-11-13T09:31:06.188397Z",
     "start_time": "2025-11-13T09:31:06.180249Z"
    }
   },
   "cell_type": "code",
   "source": [
    "missing_genes = set(gene_list3[\"Symbol\"]) - set(cnv_processed.columns)\n",
    "missing_genes"
   ],
   "id": "f5dad5ace2bc08e",
   "outputs": [
    {
     "data": {
      "text/plain": [
       "{'AARD', 'AP5M1', 'DRAXIN', 'MTARC2', 'POLR1G', 'STPG1', 'TVP23A'}"
      ]
     },
     "execution_count": 20,
     "metadata": {},
     "output_type": "execute_result"
    }
   ],
   "execution_count": 20
  },
  {
   "metadata": {
    "ExecuteTime": {
     "end_time": "2025-11-13T09:31:07.347174Z",
     "start_time": "2025-11-13T09:31:07.146056Z"
    }
   },
   "cell_type": "code",
   "source": "cnv_processed.to_csv('cnv/copy_number_variation_gistic.csv', index=False)",
   "id": "b306d69b230fb52e",
   "outputs": [],
   "execution_count": 21
  },
  {
   "metadata": {},
   "cell_type": "markdown",
   "source": "## Mutation",
   "id": "161b7078473dfd53"
  },
  {
   "metadata": {
    "ExecuteTime": {
     "end_time": "2025-11-13T09:31:09.974588Z",
     "start_time": "2025-11-13T09:31:09.779493Z"
    }
   },
   "cell_type": "code",
   "source": [
    "mut = pd.read_csv('mutation/SNVsSamples.txt', sep='\\t', index_col=0)\n",
    "mut.head()"
   ],
   "id": "f0ae40754e735e2c",
   "outputs": [
    {
     "data": {
      "text/plain": [
       "                  ID       VAF  Depth             Location  Chrom  \\\n",
       "Genotype                                                            \n",
       "0/1       CAMBMT1-T2  0.250000      8  chr10:100018799_C/A  chr10   \n",
       "0/1       CAMBMT1-T2  0.250000      8  chr10:100018802_C/T  chr10   \n",
       "0/1         HCI009-T  0.194444     36  chr10:100143399_C/T  chr10   \n",
       "1/1        HCI004-X3  1.000000     10  chr10:100174921_A/C  chr10   \n",
       "0/1          AB551-T  0.222222      9  chr10:100174935_C/A  chr10   \n",
       "\n",
       "                  Pos Ref Alt GERMLINE.1000G               Type  \\\n",
       "Genotype                                                          \n",
       "0/1       100018799.0   C   A             NO  nonsynonymous SNV   \n",
       "0/1       100018802.0   C   T             NO     synonymous SNV   \n",
       "0/1       100143399.0   C   T             NO                NaN   \n",
       "1/1       100174921.0   A   C             NO                NaN   \n",
       "0/1       100174935.0   C   A             NO                NaN   \n",
       "\n",
       "                                            Exon GenomicPos  \\\n",
       "Genotype                                                      \n",
       "0/1       LOXL4:NM_032211:exon6:c.G888T:p.K296N,     exonic   \n",
       "0/1       LOXL4:NM_032211:exon6:c.G885A:p.P295P,     exonic   \n",
       "0/1                                          NaN       UTR3   \n",
       "1/1                                          NaN       UTR5   \n",
       "0/1                                          NaN       UTR5   \n",
       "\n",
       "                                Symbol Repetitive GERMLINE.DBSNP  \n",
       "Genotype                                                          \n",
       "0/1                              LOXL4         NO             NO  \n",
       "0/1                              LOXL4         NO             NO  \n",
       "0/1       PYROXD2(NM_032709:c.*156G>A)         NO             NO  \n",
       "1/1        PYROXD2(NM_032709:c.-29T>G)         NO             NO  \n",
       "0/1        PYROXD2(NM_032709:c.-43G>T)         NO             NO  "
      ],
      "text/html": [
       "<div>\n",
       "<style scoped>\n",
       "    .dataframe tbody tr th:only-of-type {\n",
       "        vertical-align: middle;\n",
       "    }\n",
       "\n",
       "    .dataframe tbody tr th {\n",
       "        vertical-align: top;\n",
       "    }\n",
       "\n",
       "    .dataframe thead th {\n",
       "        text-align: right;\n",
       "    }\n",
       "</style>\n",
       "<table border=\"1\" class=\"dataframe\">\n",
       "  <thead>\n",
       "    <tr style=\"text-align: right;\">\n",
       "      <th></th>\n",
       "      <th>ID</th>\n",
       "      <th>VAF</th>\n",
       "      <th>Depth</th>\n",
       "      <th>Location</th>\n",
       "      <th>Chrom</th>\n",
       "      <th>Pos</th>\n",
       "      <th>Ref</th>\n",
       "      <th>Alt</th>\n",
       "      <th>GERMLINE.1000G</th>\n",
       "      <th>Type</th>\n",
       "      <th>Exon</th>\n",
       "      <th>GenomicPos</th>\n",
       "      <th>Symbol</th>\n",
       "      <th>Repetitive</th>\n",
       "      <th>GERMLINE.DBSNP</th>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>Genotype</th>\n",
       "      <th></th>\n",
       "      <th></th>\n",
       "      <th></th>\n",
       "      <th></th>\n",
       "      <th></th>\n",
       "      <th></th>\n",
       "      <th></th>\n",
       "      <th></th>\n",
       "      <th></th>\n",
       "      <th></th>\n",
       "      <th></th>\n",
       "      <th></th>\n",
       "      <th></th>\n",
       "      <th></th>\n",
       "      <th></th>\n",
       "    </tr>\n",
       "  </thead>\n",
       "  <tbody>\n",
       "    <tr>\n",
       "      <th>0/1</th>\n",
       "      <td>CAMBMT1-T2</td>\n",
       "      <td>0.250000</td>\n",
       "      <td>8</td>\n",
       "      <td>chr10:100018799_C/A</td>\n",
       "      <td>chr10</td>\n",
       "      <td>100018799.0</td>\n",
       "      <td>C</td>\n",
       "      <td>A</td>\n",
       "      <td>NO</td>\n",
       "      <td>nonsynonymous SNV</td>\n",
       "      <td>LOXL4:NM_032211:exon6:c.G888T:p.K296N,</td>\n",
       "      <td>exonic</td>\n",
       "      <td>LOXL4</td>\n",
       "      <td>NO</td>\n",
       "      <td>NO</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>0/1</th>\n",
       "      <td>CAMBMT1-T2</td>\n",
       "      <td>0.250000</td>\n",
       "      <td>8</td>\n",
       "      <td>chr10:100018802_C/T</td>\n",
       "      <td>chr10</td>\n",
       "      <td>100018802.0</td>\n",
       "      <td>C</td>\n",
       "      <td>T</td>\n",
       "      <td>NO</td>\n",
       "      <td>synonymous SNV</td>\n",
       "      <td>LOXL4:NM_032211:exon6:c.G885A:p.P295P,</td>\n",
       "      <td>exonic</td>\n",
       "      <td>LOXL4</td>\n",
       "      <td>NO</td>\n",
       "      <td>NO</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>0/1</th>\n",
       "      <td>HCI009-T</td>\n",
       "      <td>0.194444</td>\n",
       "      <td>36</td>\n",
       "      <td>chr10:100143399_C/T</td>\n",
       "      <td>chr10</td>\n",
       "      <td>100143399.0</td>\n",
       "      <td>C</td>\n",
       "      <td>T</td>\n",
       "      <td>NO</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>UTR3</td>\n",
       "      <td>PYROXD2(NM_032709:c.*156G&gt;A)</td>\n",
       "      <td>NO</td>\n",
       "      <td>NO</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>1/1</th>\n",
       "      <td>HCI004-X3</td>\n",
       "      <td>1.000000</td>\n",
       "      <td>10</td>\n",
       "      <td>chr10:100174921_A/C</td>\n",
       "      <td>chr10</td>\n",
       "      <td>100174921.0</td>\n",
       "      <td>A</td>\n",
       "      <td>C</td>\n",
       "      <td>NO</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>UTR5</td>\n",
       "      <td>PYROXD2(NM_032709:c.-29T&gt;G)</td>\n",
       "      <td>NO</td>\n",
       "      <td>NO</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>0/1</th>\n",
       "      <td>AB551-T</td>\n",
       "      <td>0.222222</td>\n",
       "      <td>9</td>\n",
       "      <td>chr10:100174935_C/A</td>\n",
       "      <td>chr10</td>\n",
       "      <td>100174935.0</td>\n",
       "      <td>C</td>\n",
       "      <td>A</td>\n",
       "      <td>NO</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>UTR5</td>\n",
       "      <td>PYROXD2(NM_032709:c.-43G&gt;T)</td>\n",
       "      <td>NO</td>\n",
       "      <td>NO</td>\n",
       "    </tr>\n",
       "  </tbody>\n",
       "</table>\n",
       "</div>"
      ]
     },
     "execution_count": 22,
     "metadata": {},
     "output_type": "execute_result"
    }
   ],
   "execution_count": 22
  },
  {
   "metadata": {
    "ExecuteTime": {
     "end_time": "2025-11-13T09:31:10.730695Z",
     "start_time": "2025-11-13T09:31:10.699295Z"
    }
   },
   "cell_type": "code",
   "source": [
    "# we only want mutation yes/no per gene\n",
    "mut = mut[['ID', 'Symbol']].drop_duplicates()"
   ],
   "id": "7272cfe92b875249",
   "outputs": [],
   "execution_count": 23
  },
  {
   "metadata": {
    "ExecuteTime": {
     "end_time": "2025-11-13T09:31:11.060169Z",
     "start_time": "2025-11-13T09:31:11.043128Z"
    }
   },
   "cell_type": "code",
   "source": [
    "# filter out the unused cell lines\n",
    "mut_key = pd.Series(list(mut['ID'].unique())).apply(normalize_id)\n",
    "\n",
    "# find matches\n",
    "matches = pd.merge(\n",
    "    pd.DataFrame({'resp': all_resp_samples, 'key': resp_key}),\n",
    "    pd.DataFrame({'micro': list(mut['ID'].unique()), 'key': mut_key}),\n",
    "    on='key',\n",
    "    how='left'\n",
    ")"
   ],
   "id": "d6fd904eb7c61b6d",
   "outputs": [],
   "execution_count": 24
  },
  {
   "metadata": {
    "ExecuteTime": {
     "end_time": "2025-11-13T09:31:11.711449Z",
     "start_time": "2025-11-13T09:31:11.675154Z"
    }
   },
   "cell_type": "code",
   "source": [
    "mapping = {\n",
    "    \"HCI001-X4\": \"HCI001-X4C\",\n",
    "    \"HCI001-X5\": \"HCI001-X5C\",\n",
    "    # no microarray match for \"HCI002-X2C\", \"HCI002-X2CR\"\n",
    "    \"HCI002-X3\": \"HCI002-X3C\",\n",
    "    \"HCI002-X6\": \"HCI002-X6C\",\n",
    "    \"HCI005-X2\": \"HCI005-X2C\",\n",
    "    \"HCI008-X1\": \"HCI008-X1C\",\n",
    "    \"HCI009-X1R\": \"HCI009-X1C\", # might be a bad match?\n",
    "    \"HCI009-X4\": \"HCI009-X4C\",\n",
    "    \"HCI009-X7\": \"HCI009-X7C\",\n",
    "    \"HCI010-X2\": \"HCI010-X2C\",\n",
    "    \"HCI010-X2R\": \"HCI010-X2CR\",\n",
    "    \"HCI011-X1\": \"HCI011-X1C\",\n",
    "    \"IC007-X0\": \"IC007-X0C\",\n",
    "    \"STG139-X13\": \"STG139-X13C\",\n",
    "    # no matches for \"STG139-X9C\", \"STG139M-X4C\n",
    "    \"STG139M-X5\": \"STG139M-X5C\",\n",
    "    \"STG143-X2\": \"STG143-X2C\",\n",
    "    \"STG195-X3\": \"STG195-X3C\",\n",
    "    \"STG201-X3\": \"STG201-X3C\",\n",
    "    \"STG282-X1\": \"STG282-X1C\",\n",
    "    \"STG282-X1R\": \"STG282-X1CR\",\n",
    "    \"STG282-X1R1\": \"STG282-X1CR1\",\n",
    "    # no match for \"STG282-X3C\"\n",
    "    \"STG316-X1\": \"STG316-X1C\",\n",
    "    \"STG316-X1R\": \"STG316-X1CR\",\n",
    "    \"STG335-X1\": \"STG335-X1C\",\n",
    "    # no matches for \"STG335-X2C\", \"VHIO098-X1C\", \"VHIO169-X11C\", \"VHIO169-X3C\",\n",
    "    # \"VHIO179-X1C\", \"VHIO244-X1C\", \"VHIO244-X1CR\", \"VHIO244-X2C\"\n",
    "}\n",
    "# missing: 13/37 -> 24 matches\n",
    "mut[\"cell_line_name\"] = mut[\"ID\"].apply(lambda x: mapping.get(x, pd.NA))\n",
    "\n",
    "print(f'Intersection {len(set(responses[\"cell_line_name\"]).intersection(set(mut[\"cell_line_name\"])))}')\n",
    "# only keep intersecting columns\n",
    "mut = mut[~pd.isna(mut[\"cell_line_name\"])]\n",
    "mut = mut.drop(columns=['ID'])\n",
    "mut.head()"
   ],
   "id": "22cadb29fff24b65",
   "outputs": [
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "Intersection 24\n"
     ]
    },
    {
     "data": {
      "text/plain": [
       "                                                     Symbol cell_line_name\n",
       "Genotype                                                                  \n",
       "0/1                               COX15(NM_078470:c.*18T>C)      IC007-X0C\n",
       "0/1       COX15(NM_078470:exon6:c.583-1G>C,NM_004376:exo...      IC007-X0C\n",
       "1/1                                                    CUTC    STG316-X1CR\n",
       "1/1                                                    CUTC     STG316-X1C\n",
       "0/1                                                   ABCC2      IC007-X0C"
      ],
      "text/html": [
       "<div>\n",
       "<style scoped>\n",
       "    .dataframe tbody tr th:only-of-type {\n",
       "        vertical-align: middle;\n",
       "    }\n",
       "\n",
       "    .dataframe tbody tr th {\n",
       "        vertical-align: top;\n",
       "    }\n",
       "\n",
       "    .dataframe thead th {\n",
       "        text-align: right;\n",
       "    }\n",
       "</style>\n",
       "<table border=\"1\" class=\"dataframe\">\n",
       "  <thead>\n",
       "    <tr style=\"text-align: right;\">\n",
       "      <th></th>\n",
       "      <th>Symbol</th>\n",
       "      <th>cell_line_name</th>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>Genotype</th>\n",
       "      <th></th>\n",
       "      <th></th>\n",
       "    </tr>\n",
       "  </thead>\n",
       "  <tbody>\n",
       "    <tr>\n",
       "      <th>0/1</th>\n",
       "      <td>COX15(NM_078470:c.*18T&gt;C)</td>\n",
       "      <td>IC007-X0C</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>0/1</th>\n",
       "      <td>COX15(NM_078470:exon6:c.583-1G&gt;C,NM_004376:exo...</td>\n",
       "      <td>IC007-X0C</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>1/1</th>\n",
       "      <td>CUTC</td>\n",
       "      <td>STG316-X1CR</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>1/1</th>\n",
       "      <td>CUTC</td>\n",
       "      <td>STG316-X1C</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>0/1</th>\n",
       "      <td>ABCC2</td>\n",
       "      <td>IC007-X0C</td>\n",
       "    </tr>\n",
       "  </tbody>\n",
       "</table>\n",
       "</div>"
      ]
     },
     "execution_count": 25,
     "metadata": {},
     "output_type": "execute_result"
    }
   ],
   "execution_count": 25
  },
  {
   "metadata": {
    "ExecuteTime": {
     "end_time": "2025-11-13T09:31:15.304840Z",
     "start_time": "2025-11-13T09:31:15.284913Z"
    }
   },
   "cell_type": "code",
   "source": [
    "# drop everything in Symbol that comes after (\n",
    "mut['Symbol'] = mut['Symbol'].str.replace(r'\\(.*', '', regex=True)\n",
    "mut = mut.drop_duplicates()"
   ],
   "id": "ac05c181d99216df",
   "outputs": [],
   "execution_count": 26
  },
  {
   "metadata": {
    "ExecuteTime": {
     "end_time": "2025-11-13T09:31:16.082092Z",
     "start_time": "2025-11-13T09:31:15.845154Z"
    }
   },
   "cell_type": "code",
   "source": [
    "wide_mut_df = (\n",
    "    mut.assign(value=True)\n",
    "      .pivot(index='cell_line_name', columns='Symbol', values='value')\n",
    "      .fillna(False)\n",
    "      .astype(bool)\n",
    ")\n",
    "wide_mut_df.iloc[0:5, 0:5]"
   ],
   "id": "67b642ba6ef64fcf",
   "outputs": [
    {
     "name": "stderr",
     "output_type": "stream",
     "text": [
      "/var/folders/3x/f8j9tddj7flfxt9zx1gkws1m0000gn/T/ipykernel_69940/142638849.py:4: FutureWarning: Downcasting object dtype arrays on .fillna, .ffill, .bfill is deprecated and will change in a future version. Call result.infer_objects(copy=False) instead. To opt-in to the future behavior, set `pd.set_option('future.no_silent_downcasting', True)`\n",
      "  .fillna(False)\n"
     ]
    },
    {
     "data": {
      "text/plain": [
       "Symbol           A1BG  A1BG-AS1   A1CF  A2ML1  A3GALT2\n",
       "cell_line_name                                        \n",
       "HCI001-X4C      False     False   True  False    False\n",
       "HCI001-X5C      False     False   True  False    False\n",
       "HCI002-X3C      False     False  False  False    False\n",
       "HCI002-X6C      False     False  False  False    False\n",
       "HCI005-X2C      False     False  False  False    False"
      ],
      "text/html": [
       "<div>\n",
       "<style scoped>\n",
       "    .dataframe tbody tr th:only-of-type {\n",
       "        vertical-align: middle;\n",
       "    }\n",
       "\n",
       "    .dataframe tbody tr th {\n",
       "        vertical-align: top;\n",
       "    }\n",
       "\n",
       "    .dataframe thead th {\n",
       "        text-align: right;\n",
       "    }\n",
       "</style>\n",
       "<table border=\"1\" class=\"dataframe\">\n",
       "  <thead>\n",
       "    <tr style=\"text-align: right;\">\n",
       "      <th>Symbol</th>\n",
       "      <th>A1BG</th>\n",
       "      <th>A1BG-AS1</th>\n",
       "      <th>A1CF</th>\n",
       "      <th>A2ML1</th>\n",
       "      <th>A3GALT2</th>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>cell_line_name</th>\n",
       "      <th></th>\n",
       "      <th></th>\n",
       "      <th></th>\n",
       "      <th></th>\n",
       "      <th></th>\n",
       "    </tr>\n",
       "  </thead>\n",
       "  <tbody>\n",
       "    <tr>\n",
       "      <th>HCI001-X4C</th>\n",
       "      <td>False</td>\n",
       "      <td>False</td>\n",
       "      <td>True</td>\n",
       "      <td>False</td>\n",
       "      <td>False</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>HCI001-X5C</th>\n",
       "      <td>False</td>\n",
       "      <td>False</td>\n",
       "      <td>True</td>\n",
       "      <td>False</td>\n",
       "      <td>False</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>HCI002-X3C</th>\n",
       "      <td>False</td>\n",
       "      <td>False</td>\n",
       "      <td>False</td>\n",
       "      <td>False</td>\n",
       "      <td>False</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>HCI002-X6C</th>\n",
       "      <td>False</td>\n",
       "      <td>False</td>\n",
       "      <td>False</td>\n",
       "      <td>False</td>\n",
       "      <td>False</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>HCI005-X2C</th>\n",
       "      <td>False</td>\n",
       "      <td>False</td>\n",
       "      <td>False</td>\n",
       "      <td>False</td>\n",
       "      <td>False</td>\n",
       "    </tr>\n",
       "  </tbody>\n",
       "</table>\n",
       "</div>"
      ]
     },
     "execution_count": 27,
     "metadata": {},
     "output_type": "execute_result"
    }
   ],
   "execution_count": 27
  },
  {
   "metadata": {
    "ExecuteTime": {
     "end_time": "2025-11-13T09:31:18.495875Z",
     "start_time": "2025-11-13T09:31:18.486488Z"
    }
   },
   "cell_type": "code",
   "source": [
    "print(f'Missing landmark genes: {len(set(gene_list[\"Symbol\"]) - set(wide_mut_df.columns))}/{len(gene_list)}')\n",
    "print(f'Missing drug target genes: {len(set(gene_list2[\"Symbol\"]) - set(wide_mut_df.columns))}/{len(gene_list2)}')\n",
    "print(f'Missing paccman genes: {len(set(gene_list3[\"Symbol\"]) - set(wide_mut_df.columns))}/{len(gene_list3)}')"
   ],
   "id": "f8a0ed1f8826b6c6",
   "outputs": [
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "Missing landmark genes: 556/867\n",
      "Missing drug target genes: 138/242\n",
      "Missing paccman genes: 1212/1957\n"
     ]
    }
   ],
   "execution_count": 28
  },
  {
   "metadata": {},
   "cell_type": "code",
   "outputs": [],
   "execution_count": 26,
   "source": "wide_mut_df.to_csv('mutation/mutations.csv')",
   "id": "f292746b6f9b8f16"
  }
 ],
 "metadata": {
  "kernelspec": {
   "display_name": "Python 3",
   "language": "python",
   "name": "python3"
  },
  "language_info": {
   "codemirror_mode": {
    "name": "ipython",
    "version": 2
   },
   "file_extension": ".py",
   "mimetype": "text/x-python",
   "name": "python",
   "nbconvert_exporter": "python",
   "pygments_lexer": "ipython2",
   "version": "2.7.6"
  }
 },
 "nbformat": 4,
 "nbformat_minor": 5
}
